{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/pschafer/miniforge3/envs/paperqa-env/lib/python3.11/site-packages/pydantic/_internal/_config.py:345: UserWarning: Valid config keys have changed in V2:\n",
      "* 'fields' has been removed\n",
      "  warnings.warn(message, UserWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DEEPSEEK_API_KEY found\n",
      "OPENAI_API_KEY found\n"
     ]
    }
   ],
   "source": [
    "from pathlib import Path\n",
    "import logging\n",
    "from datetime import datetime\n",
    "import os\n",
    "\n",
    "from paperqa.agents.main import agent_query\n",
    "from paperqa.settings import Settings, AgentSettings, IndexSettings, AnswerSettings, ParsingSettings, PromptSettings\n",
    "\n",
    "from src.build_search_index import build_search_index, process_bibtex_and_pdfs, create_manifest_file\n",
    "from src.query_answer_index import query_answer_index\n",
    "from src.utils import pretty_print_text\n",
    "\n",
    "# Boolean variable to determine if logs should also be printed to the console\n",
    "print_logging = False\n",
    "\n",
    "# Generate a timestamp for the log file name\n",
    "timestamp = datetime.now().strftime(\"%Y-%m-%d_%H-%M-%S\")\n",
    "logging_dir = Path(\".\") / \"logs\"\n",
    "logging_dir.mkdir(exist_ok=True)\n",
    "\n",
    "# Log file path\n",
    "log_file_path = logging_dir / f\"log_{timestamp}.log\"\n",
    "\n",
    "# Configure global logging\n",
    "handlers = []\n",
    "\n",
    "# Add file handler\n",
    "file_handler = logging.FileHandler(log_file_path)\n",
    "file_handler.setLevel(logging.INFO)  # Set desired file log level\n",
    "file_formatter = logging.Formatter(\"%(asctime)s - %(levelname)s - %(message)s\")\n",
    "file_handler.setFormatter(file_formatter)\n",
    "handlers.append(file_handler)\n",
    "\n",
    "# Optionally add console handler\n",
    "if print_logging:\n",
    "    console_handler = logging.StreamHandler()\n",
    "    console_handler.setLevel(logging.INFO)  # Set desired console log level\n",
    "    console_formatter = logging.Formatter(\"%(asctime)s - %(levelname)s - %(message)s\")\n",
    "    console_handler.setFormatter(console_formatter)\n",
    "    handlers.append(console_handler)\n",
    "\n",
    "# Set up the root logger\n",
    "logging.basicConfig(level=logging.INFO, handlers=handlers)\n",
    "\n",
    "# NOTE: these are the paths that should be configured\n",
    "export_directory_name = \"MS_EXPORT\"\n",
    "project_dir = Path(\".\")\n",
    "#project_dir = Path(\"/\") / \"Users\" / \"pschafer\" / \"Projects\" / \"MS_expert\"\n",
    "\n",
    "# default paths\n",
    "data_dir = project_dir / \"data\"\n",
    "data_dir.mkdir(exist_ok=True)\n",
    "paper_directory = data_dir / export_directory_name\n",
    "index_directory = data_dir / f\"{export_directory_name}_index\"\n",
    "bibtex_file = paper_directory / f\"{export_directory_name}.bib\"\n",
    "manifest_file = data_dir / f\"{export_directory_name}_manifest.csv\"\n",
    "index_name = f\"pqa_index_{export_directory_name}\"\n",
    "\n",
    "# create manifest file from bibtex\n",
    "processed_df = process_bibtex_and_pdfs(bibtex_file=bibtex_file, paper_directory=paper_directory)\n",
    "create_manifest_file(manifest_df=processed_df, manifest_file=manifest_file)\n",
    "\n",
    "#manifest_from_bibtex(bibtex_file=bibtex_file, \n",
    "#                     paper_directory=paper_directory, \n",
    "#                     manifest_file=manifest_file)\n",
    "\n",
    "# set paperQA settings\n",
    "#default_lmm = \"gpt-4o-mini\" # smaller than default which is gpt-4o (bc cheaper)\n",
    "default_lmm = \"deepseek/deepseek-chat\" # see https://docs.litellm.ai/docs/providers/deepseek\n",
    "\n",
    "index_settings = IndexSettings(\n",
    "    name = index_name,\n",
    "    paper_directory = paper_directory,\n",
    "    manifest_file = manifest_file,\n",
    "    index_directory = index_directory,\n",
    "    use_absolute_paper_directory = False,\n",
    "    recurse_subdirectories = True,\n",
    "    concurrency = 1, # \"number of concurrent filesystem reads for indexing (probably not important anymore since I avoid calling S2)\"\n",
    ")\n",
    "\n",
    "agent_settings = AgentSettings(\n",
    "    agent_llm = default_lmm,\n",
    "    index = index_settings,\n",
    "    index_concurrency = index_settings.concurrency\n",
    ")\n",
    "\n",
    "answer_settings = AnswerSettings(\n",
    "    evidence_k = 30, # number of evidence text chunks to retrieve (default=10)\n",
    "    evidence_summary_length = \"about 200 words\", # length of evidence summary (default=\"about 100 words\")\n",
    "    answer_max_sources = 15, # max number of sources to use for answering (default=5)\n",
    "    answer_length = \"about 400 words, but can be longer\", # length of final answer (default=\"about 200 words, but can be longer\")\n",
    ")\n",
    "\n",
    "parse_settings = ParsingSettings(\n",
    "    chunk_size=5_000,\n",
    "    use_doc_details=True,\n",
    "    overlap=250\n",
    ")\n",
    "\n",
    "prompt_settings = PromptSettings()\n",
    "\n",
    "settings = Settings(\n",
    "    agent = agent_settings, \n",
    "    answer = answer_settings,\n",
    "    parsing = parse_settings,\n",
    "    prompts = prompt_settings,\n",
    "    llm=default_lmm, \n",
    "    summary_llm=default_lmm,\n",
    "    embedding=\"text-embedding-3-small\", # default\n",
    "    temperature = 0.0, # default\n",
    "    texts_index_mmr_lambda = 0.95, # Lambda MMR (default=1.0)\n",
    "    index_absolute_directory = index_settings.use_absolute_paper_directory,\n",
    "    index_directory = index_settings.index_directory,\n",
    "    index_recursively = index_settings.recurse_subdirectories,\n",
    "    manifest_file = index_settings.manifest_file,\n",
    "    paper_directory = index_settings.paper_directory,\n",
    "    verbosity = 0, # (0-3), where 3 is all LLM/Embedding calls are logged\n",
    ")\n",
    "\n",
    "# Make sure that I am using the defautl arguments where it matters\n",
    "def print_non_default_settings(settings_defined, settings_classs, settings_name):\n",
    "    print(f\"------\\n{settings_name}\")\n",
    "    for key, value in settings_defined.__dict__.items():\n",
    "        default_value = getattr(settings_classs(), key, None)\n",
    "        if value != default_value:\n",
    "            print(f\"selected: {key}: {value}\")\n",
    "            print(f\"-> default: {key}: {default_value}\")\n",
    "\n",
    "# Print non-default settings for each object\n",
    "#print_non_default_settings(index_settings, IndexSettings, \"index_settings\")\n",
    "#print_non_default_settings(agent_settings, AgentSettings, \"agent_settings\")\n",
    "#print_non_default_settings(answer_settings, AnswerSettings, \"answer_settings\")\n",
    "#print_non_default_settings(parse_settings, ParsingSettings, \"parse_settings\")\n",
    "#print_non_default_settings(prompt_settings, PromptSettings, \"prompt_settings\")\n",
    "#print_non_default_settings(settings, Settings, \"settings\")\n",
    "\n",
    "# check API_KEYS are present\n",
    "API_KEYS = [\"DEEPSEEK_API_KEY\", \"OPENAI_API_KEY\"]\n",
    "for api_key in API_KEYS:\n",
    "    if (key := os.getenv(api_key)):\n",
    "        print(f\"{api_key} found\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "PromptSettings(summary='Summarize the excerpt below to help answer a question.\\n\\nExcerpt from {citation}\\n\\n----\\n\\n{text}\\n\\n----\\n\\nQuestion: {question}\\n\\nDo not directly answer the question, instead summarize to give evidence to help answer the question. Stay detailed; report specific numbers, equations, or direct quotes (marked with quotation marks). Reply \"Not applicable\" if the excerpt is irrelevant. At the end of your response, provide an integer score from 1-10 on a newline indicating relevance to question. Do not explain your score.\\n\\nRelevant Information Summary ({summary_length}):', qa='Answer the question below with the context.\\n\\nContext (with relevance scores):\\n\\n{context}\\n\\n----\\n\\nQuestion: {question}\\n\\nWrite an answer based on the context. If the context provides insufficient information reply \"I cannot answer.\" For each part of your answer, indicate which sources most support it via citation keys at the end of sentences, like {example_citation}. Only cite from the context below and only use the valid keys. Write in the style of a Wikipedia article, with concise sentences and coherent paragraphs. The context comes from a variety of sources and is only a summary, so there may inaccuracies or ambiguities. If quotes are present and relevant, use them in the answer. This answer will go directly onto Wikipedia, so do not add any extraneous information.\\n\\nAnswer ({answer_length}):', select='Select papers that may help answer the question below. Papers are listed as $KEY: $PAPER_INFO. Return a list of keys, separated by commas. Return \"None\", if no papers are applicable. Choose papers that are relevant, from reputable sources, and timely (if the question requires timely information).\\n\\nQuestion: {question}\\n\\nPapers: {papers}\\n\\nSelected keys:', pre=None, post=None, system='Answer in a direct and concise tone. Your audience is an expert, so be highly specific. If there are ambiguous terms or acronyms, first define them.', use_json=True, summary_json='Excerpt from {citation}\\n\\n----\\n\\n{text}\\n\\n----\\n\\nQuestion: {question}\\n\\n', summary_json_system='Provide a summary of the relevant information that could help answer the question based on the excerpt. Respond with the following JSON format:\\n\\n{{\\n  \"summary\": \"...\",\\n  \"relevance_score\": \"...\"\\n}}\\n\\nwhere `summary` is relevant information from the text - {summary_length} words. `relevance_score` is an integer 1-10 for the relevance of `summary` to the question.\\n', context_outer='{context_str}\\n\\nValid Keys: {valid_keys}', context_inner='{name}: {text}\\nFrom {citation}')"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prompt_settings"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Build Search Index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index Name: pqa_index_MS_EXPORT\n",
      "Number of Indexed Files: 241\n"
     ]
    }
   ],
   "source": [
    "search_index = await build_search_index(settings=settings, bibtex_file=bibtex_file, manifest_file=manifest_file)\n",
    "assert search_index.index_name == settings.agent.index.name\n",
    "print(f\"Index Name: {search_index.index_name}\")\n",
    "print(f\"Number of Indexed Files: {len((await search_index.index_files).keys())}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Perform Queries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[92m16:27:27 - LiteLLM Router:INFO\u001b[0m: router.py:611 - Routing strategy: simple-shuffle\n",
      "\u001b[92m16:27:27 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:27:30 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:27:30 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:27:32 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:27:33 - LiteLLM Router:INFO\u001b[0m: router.py:611 - Routing strategy: simple-shuffle\n",
      "\u001b[92m16:27:33 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:27:33 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:27:33 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:27:33 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:27:39 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:27:39 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:27:39 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:27:39 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:27:40 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:27:40 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:27:40 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:27:40 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:27:44 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:27:44 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:27:44 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:27:44 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:27:45 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:27:45 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:27:45 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:27:45 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:27:50 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:27:50 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:27:51 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:27:51 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:27:51 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:27:51 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:27:52 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:27:52 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:27:55 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:27:55 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:27:57 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:27:57 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:27:58 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:27:58 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:27:58 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:27:58 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:00 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:00 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:02 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:02 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:03 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:03 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:03 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:03 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:05 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:05 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:07 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:07 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:08 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:08 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:10 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:10 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:11 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:11 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:12 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:12 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:13 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:15 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:15 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:18 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:18 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:20 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:20 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:22 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:24 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:24 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:24 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:24 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:28 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:28 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:28 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:28 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:29 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:29 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:29 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:29 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:32 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:32 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:33 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:33 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:33 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:33 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:33 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:33 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:37 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:37 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:37 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:37 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:37 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:37 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:38 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:38 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:41 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:41 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:41 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:41 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:41 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:41 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:43 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:43 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:46 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:46 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:46 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:46 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:46 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:46 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:47 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:47 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:50 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:50 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:50 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:50 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:51 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:51 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:51 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:51 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:55 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:55 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:55 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:55 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:28:56 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:57 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:28:58 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:29:00 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:29:00 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:29:01 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n",
      "\u001b[92m16:29:01 - LiteLLM Router:INFO\u001b[0m: router.py:611 - Routing strategy: simple-shuffle\n",
      "\u001b[92m16:29:01 - LiteLLM:INFO\u001b[0m: utils.py:2820 - \n",
      "LiteLLM completion() model= deepseek-chat; provider = deepseek\n",
      "\u001b[92m16:29:05 - LiteLLM Router:INFO\u001b[0m: router.py:960 - litellm.acompletion(model=deepseek/deepseek-chat)\u001b[32m 200 OK\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Question: What is the role of NPPA in heart failure\n",
      "\n",
      "\n",
      "I cannot answer. The provided context does not contain any information about\n",
      "the role of NPPA (Natriuretic Peptide A, also known as atrial natriuretic\n",
      "peptide or ANP) in heart failure. The excerpts focus exclusively on the role of\n",
      "Tumor Necrosis Factor-Alpha (TNF-a) in the central nervous system, particularly\n",
      "in autoimmune disorders, and do not mention NPPA or its involvement in\n",
      "cardiovascular conditions. For information on NPPA's role in heart failure,\n",
      "additional sources specifically addressing this topic would be required.\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "answer_response = await agent_query(\n",
    "    query=\"What is the role of NPPA in heart failure\",\n",
    "    settings=settings, rebuild_index=False\n",
    ")\n",
    "pretty_print_text(answer_response.session.formatted_answer)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Query Previous Question & Answers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of Indexed Answers: 22\n",
      "Number of Answers Matching Query: 9\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "AnswerResponse(session=PQASession(id=UUID('5c1d9ac9-b3fa-48bd-bc62-1c76e21db6b7'), question='What is the gene CD163? And how is CD163 involved in multiple sclerosis?', answer='**CD163** is a gene encoding a scavenger receptor that binds to the hemoglobin-haptoglobin complex, playing a critical role in iron uptake and metabolism within myeloid cells (MCs), particularly in the context of inflammatory and neurodegenerative diseases such as multiple sclerosis (MS) (Hofmann2023 pages 1-2, Hofmann2023 pages 10-12). CD163 is primarily expressed on perivascular macrophages in the healthy brain but is upregulated on resident microglia and infiltrating macrophages under pathological conditions, including chronic active MS lesions, HIV encephalitis, and subarachnoid hemorrhage (Hofmann2023 pages 10-12). This receptor facilitates the internalization of haptoglobin-bound hemoglobin, which is enzymatically broken down by heme oxygenase (HMOX1) to release ferrous iron (Fe2+). The iron is then either stored in ferritin or exported via ferroportin (FPN), regulated by hepcidin (HAMP) (Hofmann2023 pages 13-15).\\n\\nIn MS, CD163 is significantly upregulated on a subtype of MCs at the rims of chronic active lesions, known as paramagnetic rim lesions (PRLs), where it correlates strongly with iron-positive MCs (Hofmann2023 pages 6-7, Hofmann2023 pages 10-12). This upregulation is associated with pro-inflammatory activation, as indicated by the co-expression of markers like C1QA and HLA-DR, and is linked to compromised blood-brain barrier integrity, allowing the influx of haptoglobin-hemoglobin complexes into the central nervous system (CNS) (Hofmann2023 pages 10-12, Hofmann2023 pages 7-9). The CD163-HMOX1-HAMP pathway, which is upregulated in MCs at chronic active lesion rims, suggests a mechanism for iron uptake and regulation independent of transferrin-mediated pathways (Hofmann2023 pages 6-7).\\n\\nCD163-mediated iron uptake contributes to iron accumulation at lesion rims, a hallmark of chronic active MS lesions, and is implicated in the formation of PRLs (Hofmann2023 pages 9-10). Elevated levels of soluble CD163 (sCD163), shed by ADAM17, are observed in the cerebrospinal fluid (CSF) of MS patients with higher PRL counts, correlating with disease severity and clinical disability (Hofmann2023 pages 1-2, Hofmann2023 pages 9-10). Additionally, CD163-expressing MCs exhibit both pro-inflammatory and regulatory functions, with increased co-expression of C1QA at lesion rims and IL-10 in peri-plaque white matter, highlighting its dual role in MS pathology (Hofmann2023 pages 7-9).\\n\\nIn summary, CD163 is a key player in iron metabolism and immune regulation in MS, contributing to iron accumulation, inflammation, and lesion progression. Its soluble form, sCD163, has been proposed as a potential biomarker for MS lesion activity and clinical disability (Hofmann2023 pages 10-12, Maroto2023 pages 6-7).', has_successful_answer=True, context='Hofmann2023 pages 1-2: In the context of multiple sclerosis (MS), CD163 plays a significant role in iron uptake and metabolism within myeloid cells (MCs) at the edges of chronic lesions, known as paramagnetic rim lesions (PRLs). The study by  highlights that iron uptake in these lesions is linked to rim-associated MC subtypes expressing C1QA, which are characterized by the upregulation of CD163, HMOX1, and HAMP, and downregulation of P2RY12. CD163 is a scavenger receptor for haptoglobin-bound hemoglobin, suggesting that haptoglobin-bound hemoglobin, rather than transferrin-bound iron, is a critical source for MC-associated iron uptake in MS. The study also found that people with MS (pwMS) who have higher numbers of PRLs exhibit elevated levels of soluble CD163 (sCD163) in their cerebrospinal fluid (CSF), correlating with the number of PRLs. This indicates that CD163-mediated iron uptake via haptoglobin-bound hemoglobin is relevant in MS pathology. Additionally, while haptoglobin (Hp) genotypes did not influence PRL counts, pwMS with Hp2-1/Hp2-2 haplotypes had higher clinical disability scores compared to those with Hp1-1, suggesting a potential role of CD163 in clinical worsening.\\nFrom Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.\\n\\nHofmann2023 pages 10-12: CD163 is a gene encoding a receptor for the hemoglobin-haptoglobin complex, which plays a role in iron uptake by myeloid cells (MCs) in the context of multiple sclerosis (MS) lesions. The study highlights that CD163 is significantly upregulated on a subtype of MCs at the rims of chronic active MS lesions and strongly correlates with iron-positive MCs. This suggests that CD163-mediated iron uptake may contribute to iron accumulation at lesion rims. CD163 is typically expressed on perivascular macrophages in the healthy brain but is upregulated on resident microglia and infiltrating macrophages under pathological conditions, such as in chronic active MS lesions, HIV encephalitis, and subarachnoid hemorrhage. The upregulation of CD163 is associated with pro-inflammatory activation and iron accumulation, indicating its role in pathological iron metabolism in the CNS. Additionally, soluble CD163 (sCD163), shed by ADAM17, is proposed as a potential biomarker for MS lesion activity and clinical disability.\\nFrom Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.\\n\\nHofmann2023 pages 10-12: CD163, a hemoglobin-haptoglobin complex receptor, is significantly upregulated on a subtype of myeloid cells (MCs) at the rims of chronic active lesions in multiple sclerosis (MS). This upregulation correlates with iron-positive MCs, suggesting that CD163-mediated iron uptake plays a role in iron accumulation at lesion rims. CD163 is typically expressed on perivascular macrophages in the healthy brain but is also found on resident microglia and infiltrating macrophages under pathological conditions, including MS. The expression of CD163 is associated with pro-inflammatory activation, as indicated by HLA-DR expression, and is linked to compromised blood-brain barrier integrity, allowing the influx of haptoglobin-hemoglobin complexes into the CNS. This process may contribute to microglial activation and iron accumulation in MS lesions. Additionally, soluble CD163 (sCD163) shed by ADAM17, a metallopeptidase, has been proposed as a potential biomarker for paramagnetic rim lesions (PRLs) and clinical disability in MS.\\nFrom Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.\\n\\nHofmann2023 pages 13-15: CD163 is a high-affinity scavenger receptor found on the surface of myeloid cells (MCs) at chronic active multiple sclerosis (MS) lesion rims. It binds to haptoglobin-hemoglobin complexes, facilitating their internalization into MCs. This process is crucial for iron metabolism, as the internalized hemoglobin is enzymatically broken down by heme oxygenase (HMOX1) to release ferrous iron (Fe2+). The iron is then either stored in ferritin or exported out of the cell via ferroportin (FPN), a process regulated by hepcidin. CD163 is strongly associated with iron uptake and retention in MCs, particularly in pro-inflammatory lesion areas, and its upregulation is linked to the persistence of iron in MS lesions. The gene encoding CD163 plays a significant role in iron homeostasis and immune regulation in MS pathology.\\nFrom Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.\\n\\nHofmann2023 pages 13-15: CD163 is a high-affinity scavenger receptor involved in the iron uptake and metabolism mechanisms in iron-laden myeloid cells (MCs) at chronic active multiple sclerosis (MS) lesion rims. It binds to haptoglobin-hemoglobin complexes at the surface of lesion rim-associated MCs, leading to the internalization of these complexes. This process is followed by the enzymatic breakdown of heme to form biliverdin through heme oxygenase (HMOX1) function, and the transport of ferrous iron (Fe2+) through DMT1 (encoded by SLC11A2). The iron can either be oxidized to Fe3+ and bound to ferritin or directly exported into extracellular spaces through ferroportin (FPN, encoded by SLC40A1), which is regulated by hepcidin (encoded by HAMP). HMOX1 is strongly upregulated in CD163 co-expressing MCs at chronic active lesion edges, indicating that CD163 is actively scavenging hemoglobin, a major source of iron in MCs at iron-accumulating MS lesion edges. Additionally, the upregulation of HAMP encoding hepcidin in MCs at MS lesion areas suggests that imported iron is actively retained in MCs, preventing its release into the surrounding tissue environment.\\nFrom Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.\\n\\nHofmann2023 pages 2-2: CD163 is a gene encoding a receptor for the haptoglobin-hemoglobin complex, which is primarily expressed on perivascular macrophages in subcortical white matter areas. In the context of multiple sclerosis (MS), CD163 is selectively upregulated on a subtype of myeloid cells (MCs), such as microglia and macrophages, in active and chronic active MS lesion rim areas. The extracellular domain of membrane-bound CD163 can be shed by TACE/ADAM17, forming soluble CD163 (sCD163). Elevated levels of sCD163 in the cerebrospinal fluid (CSF) of people with MS (pwMS) have been associated with higher cortical lesion load, indicating a more severe disease course. CD163 plays a role in iron uptake mechanisms in MCs, particularly in chronic active lesions, and is linked to the formation of paramagnetic rims (PRLs) in MS. The ligands for CD163 are iron-containing hemoglobin-haptoglobin complexes, which are involved in preventing oxidative stress.\\nFrom Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.\\n\\nHofmann2023 pages 6-7: CD163 is a scavenger receptor for hemoglobin-haptoglobin complexes, which plays a significant role in iron uptake by myeloid cells (MCs) in multiple sclerosis (MS) lesions. The study highlights that CD163 is highly expressed in chronic active lesion rims (LR) and is associated with iron accumulation in these regions. It is upregulated in MCs at MS lesion rims compared to other areas, such as lesion cores (LC), normal-appearing white matter (NAWM), and control (CTRL) regions. CD163 expression correlates with other iron-related proteins, such as HMOX1 (heme oxygenase 1) and HAMP (hepcidin), forming the CD163-HMOX1-HAMP pathway, which is upregulated in MCs at chronic active lesion rims. This pathway suggests a mechanism for iron uptake and regulation in MS lesions, independent of transferrin-mediated pathways.\\nFrom Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.\\n\\nHofmann2023 pages 6-7: CD163 plays a significant role in iron uptake and accumulation in myeloid cells (MCs) at the lesion rims of chronic active multiple sclerosis (MS) lesions. The study found that CD163, a scavenger receptor for hemoglobin-haptoglobin complexes, was highly expressed in MCs at these lesion rims, correlating strongly with iron accumulation . This suggests that CD163-mediated iron uptake is a key mechanism in MS pathology. Additionally, CD163 was part of the upregulated CD163-HMOX1-HAMP pathway, which is involved in iron metabolism and heme breakdown. CD163-expressing MCs also showed increased expression of IL-10 and C1QA, indicating a potential role in modulating inflammation and immune responses in MS lesions. These findings highlight CD163 as a critical player in iron-related pathology and immune regulation in MS.\\nFrom Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.\\n\\nHofmann2023 pages 7-9: The gene CD163 encodes the haptoglobin-hemoglobin receptor, which is strongly enriched at the rims of chronic active multiple sclerosis (MS) lesions. It is present on iron-laden myeloid cells (MCs) based on morphology, as shown in representative images of iron and iron-related proteins in MS lesions. CD163 is also associated with both pro-inflammatory and regulatory functions in spatially restricted lesion border areas in MS. Specifically, CD163 co-expressing cells with the pro-inflammatory marker C1QA are increased at lesion rims compared to lesion centers, while regulatory IL10 co-expressing cells are upregulated in peri-plaque white matter. These findings suggest that CD163 plays a role in iron metabolism and immune regulation in MS lesions.\\nFrom Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.\\n\\nMaroto2023 pages 6-7: CD163 is a monocyte/macrophage-specific membrane marker that is cleaved from the surface of activated macrophages as a soluble form (sCD163) during inflammation of the central nervous system (CNS). This soluble form, sCD163, can be detected in both blood and cerebrospinal fluid (CSF). Studies have shown an upregulation of plasma sCD163 and a downregulation of membrane CD163 in multiple sclerosis (MS) patients compared to healthy controls, indicating a higher release of this biomarker in MS patients. Additionally, the sCD163 CSF/serum ratio is significantly increased in MS patients, reflecting macrophage activation in MS inflammatory lesions. These findings suggest that CD163 and its soluble form play a role in the inflammatory processes associated with MS, potentially serving as biomarkers for disease progression and macrophage activity in MS.\\nFrom Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.\\n\\nHofmann2023 pages 7-9: CD163, a haptoglobin-hemoglobin receptor, is strongly enriched at the rims of chronic active multiple sclerosis (MS) lesions and is present on iron-laden myeloid cells (MCs). These MCs exhibit strong immunoreactivity for CD163, indicating its role in iron uptake and inflammation at lesion borders. CD163 is also associated with both pro-inflammatory (C1QA) and regulatory (IL10) functions, as its co-expression with C1QA is increased at lesion rims compared to lesion centers. This suggests that CD163 plays a dual role in MS pathology, contributing to both inflammatory and regulatory processes in spatially restricted lesion areas. Additionally, CD163 expression is linked to the accumulation of iron in MCs, which is a hallmark of chronic active MS lesions.\\nFrom Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.\\n\\nHofmann2023 pages 9-10: The gene CD163 encodes the hemoglobin-haptoglobin receptor, which is involved in iron uptake pathways in myeloid cells (MCs). In the context of multiple sclerosis (MS), CD163 is implicated in the active scavenging of iron via the hemoglobin-haptoglobin pathway, contributing to iron accumulation in paramagnetic rim lesions (PRLs). The study highlights that CD163+ cells are present in chronic active MS lesions, particularly at the lesion rim, and their expression correlates with iron storage and uptake mechanisms. Additionally, higher levels of soluble CD163 (sCD163) in cerebrospinal fluid (CSF) were observed in MS patients with more PRLs, suggesting a link between CD163 activity and iron-related pathology in MS.\\nFrom Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.\\n\\nHofmann2023 pages 9-10: CD163, a hemoglobin-haptoglobin receptor, plays a significant role in iron uptake and storage in myeloid cells (MCs) within chronic active lesions in multiple sclerosis (MS). The study highlights that CD163 is involved in the active scavenging of iron, contributing to the formation of paramagnetic rim lesions (PRLs) observed in MS patients. Specifically, CD163+ cells were identified in lesion areas, and their expression correlated with iron accumulation at the rim of chronic active lesions. The study also found that MS patients with higher numbers of PRLs had elevated levels of soluble CD163 (sCD163) in cerebrospinal fluid (CSF), suggesting a link between CD163-mediated iron uptake and disease severity. Additionally, the research indicates that CD163, along with other iron import mechanisms, may be responsible for sustained iron storage in PRLs, emphasizing its role in the pathological processes underlying MS.\\nFrom Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.\\n\\nLerma2022 pages 22-24: CD163 is a gene that encodes a protein which is a marker for macrophages and is used in immunohistochemistry to identify these cells in tissue samples. In the study by Lerma-Martin et al., CD163 was used as one of the antibodies for immunohistochemical staining of snap-frozen tissue blocks from individuals diagnosed with Multiple Sclerosis (MS) and control donors. The tissue sections were stained with mouse anti-CD163 antibody (NCL-L-CD163, 1:1,000, Novocastra) to visualize the presence of macrophages in the tissue. The staining process involved fixation, blocking of endogenous peroxidase, application of the primary antibody, and development of the staining using a chromogen. This method helps in understanding the role of macrophages in the disease microenvironment of MS.\\nFrom Lerma-Martin, Celia, et al. \"Spatial Cell Type Mapping of Multiple Sclerosis Lesions.\" *bioRxiv*, 3 Nov. 2022, doi:10.1101/2022.11.03.514906. Accessed 2024.\\n\\nMaroto2023 pages 12-12: The gene CD163 is associated with the macrophage scavenger receptor, which plays a role in immune responses. In the context of multiple sclerosis (MS), CD163 has been studied for its involvement in macrophage activity and its potential as a biomarker. For instance, soluble CD163 has been identified as a marker of macrophage activity in newly diagnosed MS patients, as highlighted in studies such as . Additionally, proteolytic shedding of CD163 has been observed in MS, suggesting its role in the disease\\'s inflammatory processes. The gene\\'s expression and its soluble form are linked to immune cell recruitment and activity within the central nervous system, making it a relevant target for understanding MS pathology and potential therapeutic interventions.\\nFrom Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.\\n\\nValid Keys: Hofmann2023 pages 1-2, Hofmann2023 pages 10-12, Hofmann2023 pages 10-12, Hofmann2023 pages 13-15, Hofmann2023 pages 13-15, Hofmann2023 pages 2-2, Hofmann2023 pages 6-7, Hofmann2023 pages 6-7, Hofmann2023 pages 7-9, Maroto2023 pages 6-7, Hofmann2023 pages 7-9, Hofmann2023 pages 9-10, Hofmann2023 pages 9-10, Lerma2022 pages 22-24, Maroto2023 pages 12-12', contexts=[Context(context=\"The gene CD163 is associated with the macrophage scavenger receptor, which plays a role in immune responses. In the context of multiple sclerosis (MS), CD163 has been studied for its involvement in macrophage activity and its potential as a biomarker. For instance, soluble CD163 has been identified as a marker of macrophage activity in newly diagnosed MS patients, as highlighted in studies such as . Additionally, proteolytic shedding of CD163 has been observed in MS, suggesting its role in the disease's inflammatory processes. The gene's expression and its soluble form are linked to immune cell recruitment and activity within the central nervous system, making it a relevant target for understanding MS pathology and potential therapeutic interventions.\", text=Text(text='', name='Maroto2023 pages 12-12', doc=Doc(docname='Maroto2023', citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.', dockey='9dc57baad90f908d2c0e807487e6986a', overwrite_fields_from_metadata=True, formatted_citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.')), score=9), Context(context='CD163 is a scavenger receptor for hemoglobin-haptoglobin complexes, which plays a significant role in iron uptake by myeloid cells (MCs) in multiple sclerosis (MS) lesions. The study highlights that CD163 is highly expressed in chronic active lesion rims (LR) and is associated with iron accumulation in these regions. It is upregulated in MCs at MS lesion rims compared to other areas, such as lesion cores (LC), normal-appearing white matter (NAWM), and control (CTRL) regions. CD163 expression correlates with other iron-related proteins, such as HMOX1 (heme oxygenase 1) and HAMP (hepcidin), forming the CD163-HMOX1-HAMP pathway, which is upregulated in MCs at chronic active lesion rims. This pathway suggests a mechanism for iron uptake and regulation in MS lesions, independent of transferrin-mediated pathways.', text=Text(text='', name='Hofmann2023 pages 6-7', doc=Doc(docname='Hofmann2023', citation='Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.', dockey='10b166c9c422b5fe2254ae8fb31f1adb', overwrite_fields_from_metadata=True, formatted_citation='Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.')), score=10), Context(context='CD163 is a monocyte/macrophage-specific membrane marker that is cleaved from the surface of activated macrophages as a soluble form (sCD163) during inflammation of the central nervous system (CNS). This soluble form, sCD163, can be detected in both blood and cerebrospinal fluid (CSF). Studies, such as those by Fabriek et al., have shown an upregulation of plasma sCD163 and a downregulation of membrane CD163 in multiple sclerosis (MS) patients compared to healthy controls. This suggests a higher release of sCD163 in MS patients, indicating macrophage activation in MS inflammatory lesions. The sCD163 CSF/serum ratio is significantly increased in MS patients, reflecting this activation.', text=Text(text='', name='Maroto2023 pages 6-7', doc=Doc(docname='Maroto2023', citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.', dockey='9dc57baad90f908d2c0e807487e6986a', overwrite_fields_from_metadata=True, formatted_citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.')), score=9), Context(context='CD163 is a gene encoding a receptor for the hemoglobin-haptoglobin complex, which plays a role in iron uptake by myeloid cells (MCs) in the context of multiple sclerosis (MS) lesions. The study highlights that CD163 is significantly upregulated on a subtype of MCs at the rims of chronic active MS lesions and strongly correlates with iron-positive MCs. This suggests that CD163-mediated iron uptake may contribute to iron accumulation at lesion rims. CD163 is typically expressed on perivascular macrophages in the healthy brain but is upregulated on resident microglia and infiltrating macrophages under pathological conditions, such as in chronic active MS lesions, HIV encephalitis, and subarachnoid hemorrhage. The upregulation of CD163 is associated with pro-inflammatory activation and iron accumulation, indicating its role in pathological iron metabolism in the CNS. Additionally, soluble CD163 (sCD163), shed by ADAM17, is proposed as a potential biomarker for MS lesion activity and clinical disability.', text=Text(text='', name='Hofmann2023 pages 10-12', doc=Doc(docname='Hofmann2023', citation='Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.', dockey='10b166c9c422b5fe2254ae8fb31f1adb', overwrite_fields_from_metadata=True, formatted_citation='Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.')), score=10), Context(context='The gene CD163 encodes the hemoglobin-haptoglobin receptor, which is involved in iron uptake pathways in myeloid cells (MCs). In the context of multiple sclerosis (MS), CD163 is implicated in the active scavenging of iron via the hemoglobin-haptoglobin pathway, contributing to iron accumulation in paramagnetic rim lesions (PRLs). The study highlights that CD163+ cells are present in chronic active MS lesions, particularly at the lesion rim, and their expression correlates with iron storage and uptake mechanisms. Additionally, higher levels of soluble CD163 (sCD163) in cerebrospinal fluid (CSF) were observed in MS patients with more PRLs, suggesting a link between CD163 activity and iron-related pathology in MS.', text=Text(text='', name='Hofmann2023 pages 9-10', doc=Doc(docname='Hofmann2023', citation='Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.', dockey='10b166c9c422b5fe2254ae8fb31f1adb', overwrite_fields_from_metadata=True, formatted_citation='Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.')), score=9), Context(context='The excerpt provided does not contain any specific information about the gene CD163. It primarily lists a series of gene names and identifiers, such as JUND, RPL36AL, HLA-A, FSD1, SCN7A, GPNMB, LTA4H, BTF3, ANTKMT, YBX1, FRMD8, NIPAL3, SCAF4, AK5, ADCY3, APOC1, ASB2, IFI30, DDX54, ASAH1, AMFR, RPS7, DCST1, CTSB, RALGDS, CD68, APOE, PAIP2, RPS27, IFITM10, BRI3, CCNC, C1QC, CARD9, CTSZ, ABCA13, HLA-DPB1, ARHGAP15, RPS12, LNPK, RPS25, and others. These genes are associated with white matter lesion progression in multiple sclerosis, as indicated by the title of the article. However, there is no mention of CD163 in the provided text, and thus no direct information about its role or relevance in this context.', text=Text(text='', name='Alsema2024 pages 8-8', doc=Doc(docname='Alsema2024', citation='Alsema, Astrid M., et al. \"Spatially Resolved Gene Signatures of White Matter Lesion Progression in Multiple Sclerosis.\" *Nature Neuroscience*, vol. 27, no. 12, 2024, pp. 2341–2353. https://doi.org/10.1038/s41593-024-01765-6. Accessed 2024.', dockey='d20b9b4733a400aedae0a8d96c38f891', overwrite_fields_from_metadata=True, formatted_citation='Alsema, Astrid M., et al. \"Spatially Resolved Gene Signatures of White Matter Lesion Progression in Multiple Sclerosis.\" *Nature Neuroscience*, vol. 27, no. 12, 2024, pp. 2341–2353. https://doi.org/10.1038/s41593-024-01765-6. Accessed 2024.')), score=1), Context(context='The gene CD163 is mentioned in the context of biomarkers for inflammation and axonal degeneration/damage in patients with newly diagnosed multiple sclerosis. Specifically, the soluble CD163 CSF/serum ratio is highlighted as a potential biomarker that contributes to a biomarker panel for assessing these conditions in multiple sclerosis patients. The study referenced in the excerpt (PLoS ONE 10 (4) (2015) e0119681) suggests that CD163, along with other biomarkers, could be useful in understanding the inflammatory and degenerative processes in multiple sclerosis.', text=Text(text='', name='Maroto2023 pages 12-12', doc=Doc(docname='Maroto2023', citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.', dockey='9dc57baad90f908d2c0e807487e6986a', overwrite_fields_from_metadata=True, formatted_citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.')), score=8), Context(context='The excerpt does not provide any information about the gene CD163. Instead, it discusses various biochemical biomarkers related to multiple sclerosis (MS), such as osteopontin (OPN), tubulin, heat shock protein 70 (HSP70), and chemokine ligand 13 (CXCL13). These biomarkers are highlighted for their roles in inflammation, disease progression, and potential diagnostic and prognostic utility in MS. For example, OPN levels in cerebrospinal fluid (CSF) and blood are associated with MS progression, while CXCL13 is noted for its sensitivity in predicting neuroinflammatory activity. HSP70 is discussed in the context of its role in immune response and its elevated levels in certain MS subtypes. The text emphasizes the need for further research to establish the clinical utility of these biomarkers.', text=Text(text='', name='Maroto2023 pages 6-6', doc=Doc(docname='Maroto2023', citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.', dockey='9dc57baad90f908d2c0e807487e6986a', overwrite_fields_from_metadata=True, formatted_citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.')), score=1), Context(context=\"The excerpt does not provide a direct definition or detailed description of the gene CD163. However, it mentions 'sCD163' (soluble cluster of differentiation 163) in the context of cerebrospinal fluid (CSF) analysis. The text describes the process of analyzing CSF samples for sCD163 concentration using the ultrasensitive Ella-Simple Plex technology, an automated ELISA method. This suggests that CD163 is a gene or protein marker of interest in the study of multiple sclerosis (MS), particularly in relation to CSF biomarkers. The analysis of sCD163 is performed to assess its concentration in CSF, which may be relevant to understanding MS pathology or disease progression.\", text=Text(text='', name='Hofmann2023 pages 4-5', doc=Doc(docname='Hofmann2023', citation='Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.', dockey='10b166c9c422b5fe2254ae8fb31f1adb', overwrite_fields_from_metadata=True, formatted_citation='Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.')), score=6), Context(context='The excerpt does not provide specific information about the gene CD163. However, it discusses the roles of myeloid cell types, including microglia (MG) and macrophages (MΦ-MS), in multiple sclerosis (MS) lesions. It highlights the upregulation of proinflammatory marker genes such as CD84 and CD14, as well as the downregulation of anti-inflammatory and apoptosis-related genes like INPP5D. The study also mentions the upregulation of transcription factors like SPI1 (PU.1), which is associated with MS. While CD163 is not explicitly mentioned, the context of myeloid cell activity and inflammation in MS lesions may be indirectly relevant to understanding genes like CD163, which are often associated with macrophage function and immune responses.', text=Text(text='', name='Lerma2022 pages 12-15', doc=Doc(docname='Lerma2022', citation='Lerma-Martin, Celia, et al. \"Spatial Cell Type Mapping of Multiple Sclerosis Lesions.\" *bioRxiv*, 3 Nov. 2022, doi:10.1101/2022.11.03.514906. Accessed 2024.', dockey='31df84a215204ca9fb769ebc5ddcd0cb', overwrite_fields_from_metadata=True, formatted_citation='Lerma-Martin, Celia, et al. \"Spatial Cell Type Mapping of Multiple Sclerosis Lesions.\" *bioRxiv*, 3 Nov. 2022, doi:10.1101/2022.11.03.514906. Accessed 2024.')), score=4), Context(context='The excerpt does not provide specific information about the gene CD163. It primarily discusses gene signatures related to white matter lesion progression in multiple sclerosis, including terms such as ANXA1, HLA-DRAGPNMB, APOC1, FTL, S100A11, IGKC, and C1QA. These genes are associated with processes like neutrophil degranulation, immune response regulation, antigen processing, and blood vessel development. The excerpt also mentions pathways such as oxidative phosphorylation and mitochondrial transport chain. However, there is no direct mention or description of CD163, its function, or its relevance in the context of the study.', text=Text(text='', name='Alsema2024 pages 11-11', doc=Doc(docname='Alsema2024', citation='Alsema, Astrid M., et al. \"Spatially Resolved Gene Signatures of White Matter Lesion Progression in Multiple Sclerosis.\" *Nature Neuroscience*, vol. 27, no. 12, 2024, pp. 2341–2353. https://doi.org/10.1038/s41593-024-01765-6. Accessed 2024.', dockey='d20b9b4733a400aedae0a8d96c38f891', overwrite_fields_from_metadata=True, formatted_citation='Alsema, Astrid M., et al. \"Spatially Resolved Gene Signatures of White Matter Lesion Progression in Multiple Sclerosis.\" *Nature Neuroscience*, vol. 27, no. 12, 2024, pp. 2341–2353. https://doi.org/10.1038/s41593-024-01765-6. Accessed 2024.')), score=1), Context(context='CD163 is a high-affinity scavenger receptor found on the surface of myeloid cells (MCs) at chronic active multiple sclerosis (MS) lesion rims. It binds to haptoglobin-hemoglobin complexes, facilitating their internalization into MCs. This process is crucial for iron metabolism, as the internalized hemoglobin is enzymatically broken down by heme oxygenase (HMOX1) to release ferrous iron (Fe2+). The iron is then either stored in ferritin or exported out of the cell via ferroportin (FPN), a process regulated by hepcidin. CD163 is strongly associated with iron uptake and retention in MCs, particularly in pro-inflammatory lesion areas, and its upregulation is linked to the persistence of iron in MS lesions. The gene encoding CD163 plays a significant role in iron homeostasis and immune regulation in MS pathology.', text=Text(text='', name='Hofmann2023 pages 13-15', doc=Doc(docname='Hofmann2023', citation='Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.', dockey='10b166c9c422b5fe2254ae8fb31f1adb', overwrite_fields_from_metadata=True, formatted_citation='Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.')), score=10), Context(context='The excerpt provided does not mention the gene CD163 or provide any information related to it. Instead, it focuses on gene regulatory circuits in immune cells, particularly T helper 17 (Th17) cells, innate lymphoid cells (ILCs), and their plasticity, as well as the role of transcription factors like c-Maf and T-bet in immune cell differentiation and function. The text discusses studies on immune cell behavior, such as Th17 cells giving rise to Th1 cells in colitis, the plasticity of ILC3 to ILC1 conversion, and the regulation of immune responses by factors like IL-12, IL-23, and Notch signaling. However, none of this information is directly relevant to the gene CD163.', text=Text(text='', name='Chan2022 pages 21-22', doc=Doc(docname='Chan2022', citation='Chan-Wang Saini, Ankita, et al. \"Gene Regulatory Circuits in Innate and Adaptive Immune Cells.\" *Annual Review of Immunology*, vol. 40, 2022, pp. 387–411. Annual Reviews, https://doi.org/10.1146/annurev-immunol-101320-025949. Accessed 2024.', dockey='0ceffe93a26439b7bfc1d6461969eec9', overwrite_fields_from_metadata=True, formatted_citation='Chan-Wang Saini, Ankita, et al. \"Gene Regulatory Circuits in Innate and Adaptive Immune Cells.\" *Annual Review of Immunology*, vol. 40, 2022, pp. 387–411. Annual Reviews, https://doi.org/10.1146/annurev-immunol-101320-025949. Accessed 2024.')), score=1), Context(context='CD163 is a gene encoding a receptor for the haptoglobin-hemoglobin complex, which is primarily expressed on perivascular macrophages in subcortical white matter areas. In the context of multiple sclerosis (MS), CD163 is selectively upregulated on a subtype of myeloid cells (MCs), such as microglia and macrophages, in active and chronic active MS lesion rim areas. The extracellular domain of membrane-bound CD163 can be shed by TACE/ADAM17, forming soluble CD163 (sCD163). Elevated levels of sCD163 in the cerebrospinal fluid (CSF) of people with MS (pwMS) have been associated with higher cortical lesion load, indicating a more severe disease course. CD163 plays a role in iron uptake mechanisms in MCs, particularly in chronic active lesions, and is linked to the formation of paramagnetic rims (PRLs) in MS. The ligands for CD163 are iron-containing hemoglobin-haptoglobin complexes, which are involved in preventing oxidative stress.', text=Text(text='', name='Hofmann2023 pages 2-2', doc=Doc(docname='Hofmann2023', citation='Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.', dockey='10b166c9c422b5fe2254ae8fb31f1adb', overwrite_fields_from_metadata=True, formatted_citation='Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.')), score=10), Context(context='The excerpt does not provide specific information about the gene CD163. It primarily lists various genes and their associated counts or expressions in different white matter lesion subgroups, such as PLWM-active, PLWM-act/inact, NAWM, and CWM. Genes mentioned include ANO4, VIM, ANXA1, CD44, LGALS3, ABCC3, and others, but CD163 is not explicitly discussed. The data appears to focus on spatially resolved gene signatures related to white matter lesion progression in multiple sclerosis, with an emphasis on gene expression patterns and their statistical significance (e.g., -log10(P-adjusted)). To answer the question about CD163, additional context or a different source would be required.', text=Text(text='', name='Alsema2024 pages 9-9', doc=Doc(docname='Alsema2024', citation='Alsema, Astrid M., et al. \"Spatially Resolved Gene Signatures of White Matter Lesion Progression in Multiple Sclerosis.\" *Nature Neuroscience*, vol. 27, no. 12, 2024, pp. 2341–2353. https://doi.org/10.1038/s41593-024-01765-6. Accessed 2024.', dockey='d20b9b4733a400aedae0a8d96c38f891', overwrite_fields_from_metadata=True, formatted_citation='Alsema, Astrid M., et al. \"Spatially Resolved Gene Signatures of White Matter Lesion Progression in Multiple Sclerosis.\" *Nature Neuroscience*, vol. 27, no. 12, 2024, pp. 2341–2353. https://doi.org/10.1038/s41593-024-01765-6. Accessed 2024.')), score=2), Context(context='CD163 is a gene that encodes a protein which is a marker for macrophages and is used in immunohistochemistry to identify these cells in tissue samples. In the study by Lerma-Martin et al., CD163 was used as one of the antibodies for immunohistochemical staining of snap-frozen tissue blocks from individuals diagnosed with Multiple Sclerosis (MS) and control donors. The tissue sections were stained with mouse anti-CD163 antibody (NCL-L-CD163, 1:1,000, Novocastra) to visualize the presence of macrophages in the tissue. The staining process involved fixation, blocking of endogenous peroxidase, application of the primary antibody, and development of the staining using a chromogen. This method helps in understanding the role of macrophages in the disease microenvironment of MS.', text=Text(text='', name='Lerma2022 pages 22-24', doc=Doc(docname='Lerma2022', citation='Lerma-Martin, Celia, et al. \"Spatial Cell Type Mapping of Multiple Sclerosis Lesions.\" *bioRxiv*, 3 Nov. 2022, doi:10.1101/2022.11.03.514906. Accessed 2024.', dockey='31df84a215204ca9fb769ebc5ddcd0cb', overwrite_fields_from_metadata=True, formatted_citation='Lerma-Martin, Celia, et al. \"Spatial Cell Type Mapping of Multiple Sclerosis Lesions.\" *bioRxiv*, 3 Nov. 2022, doi:10.1101/2022.11.03.514906. Accessed 2024.')), score=9), Context(context='The excerpt mentions CD163 as a macrophage activation marker. It is referenced in the context of acute-on-chronic liver failure, where CD163 and CD206 are highlighted as markers for macrophage activation. This suggests that CD163 is a gene or protein associated with the activation state of macrophages, which are immune cells involved in inflammation and tissue repair. The study by  investigates these markers in the context of liver disease, indicating their relevance in immune responses and pathological conditions. However, the excerpt does not provide detailed information about the specific function or structure of CD163.', text=Text(text='', name='Hofmann2023 pages 17-17', doc=Doc(docname='Hofmann2023', citation='Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.', dockey='10b166c9c422b5fe2254ae8fb31f1adb', overwrite_fields_from_metadata=True, formatted_citation='Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.')), score=7), Context(context=\"The gene CD163 is mentioned in the context of a study on white matter lesion progression in multiple sclerosis. The excerpt provides a detailed analysis of various genes and their expression in different lesion types, including active and inactive lesions. CD163 is listed alongside other genes such as VIM, RPL27A, PFDN5, IFI30, EEF1B2, FCER1G, LYZ, EEF1G, RSL24D1, and RPL36ALHLA−A. The study involves spatial resolution of gene signatures and includes data from multiple samples, such as sample M9 and sample A5, which are analyzed for their gene expression profiles. The context suggests that CD163 is one of the genes being studied for its role in the progression of white matter lesions in multiple sclerosis, potentially contributing to the understanding of the disease's pathology.\", text=Text(text='', name='Alsema2024 pages 8-8', doc=Doc(docname='Alsema2024', citation='Alsema, Astrid M., et al. \"Spatially Resolved Gene Signatures of White Matter Lesion Progression in Multiple Sclerosis.\" *Nature Neuroscience*, vol. 27, no. 12, 2024, pp. 2341–2353. https://doi.org/10.1038/s41593-024-01765-6. Accessed 2024.', dockey='d20b9b4733a400aedae0a8d96c38f891', overwrite_fields_from_metadata=True, formatted_citation='Alsema, Astrid M., et al. \"Spatially Resolved Gene Signatures of White Matter Lesion Progression in Multiple Sclerosis.\" *Nature Neuroscience*, vol. 27, no. 12, 2024, pp. 2341–2353. https://doi.org/10.1038/s41593-024-01765-6. Accessed 2024.')), score=8), Context(context='The excerpt does not provide any information about the gene CD163. Instead, it discusses various biomarkers related to multiple sclerosis (MS), such as S100β, Chitinase-3-like 1 (CHI3L1), CD5 + B cells, and osteopontin (OPN). These biomarkers are explored for their potential roles in MS diagnosis, prognosis, and treatment efficacy. For example, S100β levels are noted to increase in MS patients and may serve as a biomarker for treatment efficacy. CHI3L1 is highlighted as a potential biomarker for monitoring MS progression and treatment response. CD5 + B cells are associated with MS development and disease activity. Osteopontin is mentioned in the context of inflammation and autoimmune disorders, including MS. However, there is no mention of CD163 in the provided text.', text=Text(text='', name='Maroto2023 pages 5-6', doc=Doc(docname='Maroto2023', citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.', dockey='9dc57baad90f908d2c0e807487e6986a', overwrite_fields_from_metadata=True, formatted_citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.')), score=1), Context(context='The provided excerpt does not contain any information about the gene CD163. Instead, it discusses various biochemical biomarkers and their roles in multiple sclerosis (MS). For example, it mentions intrathecally produced CXCL13 as a predictive biomarker in MS, the association of meningeal B-cell follicles with early onset and severe cortical pathology in secondary progressive MS, and the detection of ectopic B-cell follicles with germinal centers in the meninges of MS patients. Additionally, it references the role of CXCL12 chemokine and its receptors in interactions between the immune and nervous systems, as well as the potential of heat shock proteins (HSP70/HSC70) as biomarkers for inflammatory processes in MS. However, none of these details are relevant to the gene CD163.', text=Text(text='', name='Maroto2023 pages 12-12', doc=Doc(docname='Maroto2023', citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.', dockey='9dc57baad90f908d2c0e807487e6986a', overwrite_fields_from_metadata=True, formatted_citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.')), score=1), Context(context='The excerpt does not provide specific information about the gene CD163. However, it discusses the role of microglia (MG) and macrophage (MΦ-MS) subtypes in multiple sclerosis (MS) lesions, particularly their proinflammatory states and interactions with oligodendrocytes (OL). Key findings include the upregulation of proinflammatory pathways (e.g., TNFα, JAK-STAT) and hypoxia-related pathways in MG states 2 and 3, as well as enriched ligand-receptor interactions such as APOE-TREM2 and APOE-LRP1, which are associated with myelin debris clearance and lipid/apoptotic cell clearance in inflamed MS lesions. Additionally, the study highlights the transcriptomic heterogeneity of astrocyte (AS) subtypes and their roles in MS pathogenesis. While CD163 is not mentioned, the focus on myeloid cell interactions and inflammatory pathways may provide context for understanding genes like CD163, which is often associated with macrophage activity and inflammation.', text=Text(text='', name='Lerma2022 pages 15-16', doc=Doc(docname='Lerma2022', citation='Lerma-Martin, Celia, et al. \"Spatial Cell Type Mapping of Multiple Sclerosis Lesions.\" *bioRxiv*, 3 Nov. 2022, doi:10.1101/2022.11.03.514906. Accessed 2024.', dockey='31df84a215204ca9fb769ebc5ddcd0cb', overwrite_fields_from_metadata=True, formatted_citation='Lerma-Martin, Celia, et al. \"Spatial Cell Type Mapping of Multiple Sclerosis Lesions.\" *bioRxiv*, 3 Nov. 2022, doi:10.1101/2022.11.03.514906. Accessed 2024.')), score=3), Context(context='The excerpt provided does not contain any specific information about the gene CD163. It primarily discusses a histogram of genes from 10 randomly selected modules and mentions various gene ontology (GO) terms related to biological processes and molecular functions. The genes listed in the histogram and the GO terms do not include CD163. Therefore, based on the given text, there is no relevant information about the gene CD163.', text=Text(text='', name='Lin2024 pages 26-27', doc=Doc(docname='Lin2024', citation='Lin, Zaikang, Sei Chang, Aaron Zweig, Elham Azizi, and David A. Knowles. \"Interpretable Neural ODEs for Gene Regulatory Network Discovery Under Perturbations.\" *arXiv*, 2024, arXiv:2501.02409v1. Accessed 2024.', dockey='e2eec99d8fb2107941606b212a549ad9', overwrite_fields_from_metadata=True, formatted_citation='Lin, Zaikang, Sei Chang, Aaron Zweig, Elham Azizi, and David A. Knowles. \"Interpretable Neural ODEs for Gene Regulatory Network Discovery Under Perturbations.\" *arXiv*, 2024, arXiv:2501.02409v1. Accessed 2024.')), score=1), Context(context=\"The excerpt does not provide specific information about the gene CD163. However, it discusses the mapping of MS susceptibility genes and the identification of MS-blood vessel (MS-BV) spots in white matter (WM) lesion samples. The study identifies 12 MS-BV signature genes by overlapping BV cluster marker genes from 4 WM lesion samples. The analysis includes spatial plots mapping MS susceptibility genes onto spatial clusters and the identification of blood vessel (BV) clusters in WM lesion samples. The study also examines the percentage of positive MS-BV spots per subgroup for active and act/inact lesions, with statistical significance determined using repeated measures ANOVA and linear mixed models. While the gene CD163 is not mentioned, the study's focus on gene signatures and blood vessel analysis in MS lesions may provide a context for understanding the role of specific genes in MS pathology.\", text=Text(text='', name='Alsema2024 pages 28-29', doc=Doc(docname='Alsema2024', citation='Alsema, Astrid M., et al. \"Spatially Resolved Gene Signatures of White Matter Lesion Progression in Multiple Sclerosis.\" *Nature Neuroscience*, vol. 27, no. 12, 2024, pp. 2341–2353. https://doi.org/10.1038/s41593-024-01765-6. Accessed 2024.', dockey='d20b9b4733a400aedae0a8d96c38f891', overwrite_fields_from_metadata=True, formatted_citation='Alsema, Astrid M., et al. \"Spatially Resolved Gene Signatures of White Matter Lesion Progression in Multiple Sclerosis.\" *Nature Neuroscience*, vol. 27, no. 12, 2024, pp. 2341–2353. https://doi.org/10.1038/s41593-024-01765-6. Accessed 2024.')), score=3), Context(context='The excerpt does not mention the gene CD163. Instead, it focuses on the gene regulatory circuits in innate and adaptive immune cells, particularly CD8+ T cells and NK cells. It discusses the epigenetic mechanisms, such as histone modifications (e.g., H3K27me3, H3K4me3) and transcription factors (e.g., RUNX3, TCF1, EOMES, T-bet), that regulate the differentiation and function of these cells. The text also highlights the roles of enhancers and chromatin accessibility in shaping memory and effector phenotypes in CD8+ T cells and NK cells. Additionally, it mentions the interplay between TCF1 and BLIMP1 in balancing self-renewal and effector functions in lymphoid populations.', text=Text(text='', name='Chan2022 pages 12-12', doc=Doc(docname='Chan2022', citation='Chan-Wang Saini, Ankita, et al. \"Gene Regulatory Circuits in Innate and Adaptive Immune Cells.\" *Annual Review of Immunology*, vol. 40, 2022, pp. 387–411. Annual Reviews, https://doi.org/10.1146/annurev-immunol-101320-025949. Accessed 2024.', dockey='0ceffe93a26439b7bfc1d6461969eec9', overwrite_fields_from_metadata=True, formatted_citation='Chan-Wang Saini, Ankita, et al. \"Gene Regulatory Circuits in Innate and Adaptive Immune Cells.\" *Annual Review of Immunology*, vol. 40, 2022, pp. 387–411. Annual Reviews, https://doi.org/10.1146/annurev-immunol-101320-025949. Accessed 2024.')), score=1), Context(context='The excerpt does not provide specific information about the gene CD163. However, it details the methodology used in the study, including the analysis of gene expression in white matter lesions in multiple sclerosis. The study involves the use of Nanostring GeoMx Digital Spatial Profiler for sample preparation, data analysis using tools like GeoMxTools v.3.0.1, and statistical comparisons of gene set module scores. The analysis includes hierarchical clustering of the top 1,000 most variable genes and pathway and Gene Ontology enrichment using Metascape. While the excerpt does not mention CD163 directly, the described methods could potentially be applied to study CD163 if it were among the genes analyzed in the context of white matter lesion progression in multiple sclerosis.', text=Text(text='', name='Alsema2024 pages 16-17', doc=Doc(docname='Alsema2024', citation='Alsema, Astrid M., et al. \"Spatially Resolved Gene Signatures of White Matter Lesion Progression in Multiple Sclerosis.\" *Nature Neuroscience*, vol. 27, no. 12, 2024, pp. 2341–2353. https://doi.org/10.1038/s41593-024-01765-6. Accessed 2024.', dockey='d20b9b4733a400aedae0a8d96c38f891', overwrite_fields_from_metadata=True, formatted_citation='Alsema, Astrid M., et al. \"Spatially Resolved Gene Signatures of White Matter Lesion Progression in Multiple Sclerosis.\" *Nature Neuroscience*, vol. 27, no. 12, 2024, pp. 2341–2353. https://doi.org/10.1038/s41593-024-01765-6. Accessed 2024.')), score=3), Context(context='The excerpt does not provide any information about the gene CD163. Instead, it discusses the etiology and genetic factors associated with Multiple Sclerosis (MS), including the role of HLA II alleles, such as HLA DRB1*1501-DQB1*0602, and other genetic risk factors like interleukin-2 (IL-2) receptor alpha gene and interleukin-7 (IL-7) receptor alpha gene. It also mentions the involvement of pro-inflammatory cytokines, such as IFN-γ, TNF-α, and IL-17, in the activation of macrophages and microglia, which contribute to the inflammatory processes in MS. Additionally, the text highlights the importance of vitamin D levels and the use of McDonald criteria for MS diagnosis.', text=Text(text='', name='Maroto2023 pages 2-2', doc=Doc(docname='Maroto2023', citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.', dockey='9dc57baad90f908d2c0e807487e6986a', overwrite_fields_from_metadata=True, formatted_citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.')), score=1), Context(context='The excerpt provided does not contain specific information about the gene CD163. Instead, it lists a variety of other genes such as LCP2, FAM43A, ARRB2, GALNT15, NPM1, H3C12, MVB12B, NACA, and others, along with their spatial relationships and z-scores in the context of white matter lesion progression in multiple sclerosis. The focus of the text is on the spatial resolution and gene signatures associated with lesion progression, but it does not mention CD163 or its role in this context. Therefore, the excerpt is not directly relevant to the question about CD163.', text=Text(text='', name='Alsema2024 pages 31-31', doc=Doc(docname='Alsema2024', citation='Alsema, Astrid M., et al. \"Spatially Resolved Gene Signatures of White Matter Lesion Progression in Multiple Sclerosis.\" *Nature Neuroscience*, vol. 27, no. 12, 2024, pp. 2341–2353. https://doi.org/10.1038/s41593-024-01765-6. Accessed 2024.', dockey='d20b9b4733a400aedae0a8d96c38f891', overwrite_fields_from_metadata=True, formatted_citation='Alsema, Astrid M., et al. \"Spatially Resolved Gene Signatures of White Matter Lesion Progression in Multiple Sclerosis.\" *Nature Neuroscience*, vol. 27, no. 12, 2024, pp. 2341–2353. https://doi.org/10.1038/s41593-024-01765-6. Accessed 2024.')), score=1), Context(context='The gene CD163 encodes the haptoglobin-hemoglobin receptor, which is strongly enriched at the rims of chronic active multiple sclerosis (MS) lesions. It is present on iron-laden myeloid cells (MCs) based on morphology, as shown in representative images of iron and iron-related proteins in MS lesions. CD163 is also associated with both pro-inflammatory and regulatory functions in spatially restricted lesion border areas in MS. Specifically, CD163 co-expressing cells with the pro-inflammatory marker C1QA are increased at lesion rims compared to lesion centers, while regulatory IL10 co-expressing cells are upregulated in peri-plaque white matter. These findings suggest that CD163 plays a role in iron metabolism and immune regulation in MS lesions.', text=Text(text='', name='Hofmann2023 pages 7-9', doc=Doc(docname='Hofmann2023', citation='Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.', dockey='10b166c9c422b5fe2254ae8fb31f1adb', overwrite_fields_from_metadata=True, formatted_citation='Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.')), score=10), Context(context='The provided excerpt does not contain any information about the gene CD163. Instead, it lists a series of gene names and identifiers, such as SLC14A1, SCD, OLIG1, PPP1R14A, ATP1B1, PRNP, and others, which are associated with white matter lesion progression in multiple sclerosis. These genes are part of a spatially resolved gene signature study, but no specific details about their functions or relevance to CD163 are provided. The excerpt appears to be a list of gene names without additional context or explanation.', text=Text(text='', name='Alsema2024 pages 29-29', doc=Doc(docname='Alsema2024', citation='Alsema, Astrid M., et al. \"Spatially Resolved Gene Signatures of White Matter Lesion Progression in Multiple Sclerosis.\" *Nature Neuroscience*, vol. 27, no. 12, 2024, pp. 2341–2353. https://doi.org/10.1038/s41593-024-01765-6. Accessed 2024.', dockey='d20b9b4733a400aedae0a8d96c38f891', overwrite_fields_from_metadata=True, formatted_citation='Alsema, Astrid M., et al. \"Spatially Resolved Gene Signatures of White Matter Lesion Progression in Multiple Sclerosis.\" *Nature Neuroscience*, vol. 27, no. 12, 2024, pp. 2341–2353. https://doi.org/10.1038/s41593-024-01765-6. Accessed 2024.')), score=1), Context(context='The excerpt provided does not mention the gene CD163. Instead, it focuses on gene regulatory circuits in innate and adaptive immune cells, discussing various transcription factors, cytokines, and immune cell differentiation processes. Key topics include the roles of ZEB2, Tcf1, T-bet, eomesodermin, and Blimp1 in CD8+ T cell differentiation and memory formation, as well as the epigenetic control of immune memory and the adaptive features of natural killer (NK) cells. The text also highlights the involvement of transcription factors like IRF8, Runx, and Zbtb32 in regulating immune responses. However, there is no direct or indirect reference to CD163 or its function in this context.', text=Text(text='', name='Chan2022 pages 23-24', doc=Doc(docname='Chan2022', citation='Chan-Wang Saini, Ankita, et al. \"Gene Regulatory Circuits in Innate and Adaptive Immune Cells.\" *Annual Review of Immunology*, vol. 40, 2022, pp. 387–411. Annual Reviews, https://doi.org/10.1146/annurev-immunol-101320-025949. Accessed 2024.', dockey='0ceffe93a26439b7bfc1d6461969eec9', overwrite_fields_from_metadata=True, formatted_citation='Chan-Wang Saini, Ankita, et al. \"Gene Regulatory Circuits in Innate and Adaptive Immune Cells.\" *Annual Review of Immunology*, vol. 40, 2022, pp. 387–411. Annual Reviews, https://doi.org/10.1146/annurev-immunol-101320-025949. Accessed 2024.')), score=1), Context(context='CD163 is a monocyte/macrophage-specific membrane marker that is cleaved from the surface of activated macrophages as a soluble form (sCD163) during inflammation of the central nervous system (CNS). This soluble form, sCD163, can be detected in both blood and cerebrospinal fluid (CSF). Studies have shown an upregulation of plasma sCD163 and a downregulation of membrane CD163 in multiple sclerosis (MS) patients compared to healthy controls, indicating a higher release of this biomarker in MS patients. Additionally, the sCD163 CSF/serum ratio is significantly increased in MS patients, reflecting macrophage activation in MS inflammatory lesions. These findings suggest that CD163 and its soluble form play a role in the inflammatory processes associated with MS, potentially serving as biomarkers for disease progression and macrophage activity in MS.', text=Text(text='', name='Maroto2023 pages 6-7', doc=Doc(docname='Maroto2023', citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.', dockey='9dc57baad90f908d2c0e807487e6986a', overwrite_fields_from_metadata=True, formatted_citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.')), score=10), Context(context=\"The excerpt does not provide any specific information about the involvement of CD163 in multiple sclerosis (MS). The text primarily discusses the role of pro-inflammatory cytokines, genetic factors, and environmental triggers in the etiology of MS. It also mentions the involvement of HLA alleles, interleukin receptors, and MAPK pathways in the disease's pathogenesis. Additionally, it highlights the importance of vitamin D levels and the diagnostic criteria for MS. However, there is no mention of CD163 or its role in MS within the provided text.\", text=Text(text='', name='Maroto2023 pages 2-2', doc=Doc(docname='Maroto2023', citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.', dockey='9dc57baad90f908d2c0e807487e6986a', overwrite_fields_from_metadata=True, formatted_citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.')), score=1), Context(context='CD163, a hemoglobin-haptoglobin complex receptor, is significantly upregulated on a subtype of myeloid cells (MCs) at the rims of chronic active lesions in multiple sclerosis (MS). This upregulation correlates with iron-positive MCs, suggesting that CD163-mediated iron uptake plays a role in iron accumulation at lesion rims. CD163 is typically expressed on perivascular macrophages in the healthy brain but is also found on resident microglia and infiltrating macrophages under pathological conditions, including MS. The expression of CD163 is associated with pro-inflammatory activation, as indicated by HLA-DR expression, and is linked to compromised blood-brain barrier integrity, allowing the influx of haptoglobin-hemoglobin complexes into the CNS. This process may contribute to microglial activation and iron accumulation in MS lesions. Additionally, soluble CD163 (sCD163) shed by ADAM17, a metallopeptidase, has been proposed as a potential biomarker for paramagnetic rim lesions (PRLs) and clinical disability in MS.', text=Text(text='', name='Hofmann2023 pages 10-12', doc=Doc(docname='Hofmann2023', citation='Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.', dockey='10b166c9c422b5fe2254ae8fb31f1adb', overwrite_fields_from_metadata=True, formatted_citation='Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.')), score=10), Context(context=\"The excerpt discusses the role of CD163 in multiple sclerosis (MS), particularly as a biomarker for inflammation and axonal degeneration/damage in newly diagnosed MS patients. The soluble CD163 CSF/serum ratio is highlighted as a significant contributor to a biomarker panel for MS. This suggests that CD163 levels in cerebrospinal fluid (CSF) and serum may be indicative of inflammatory processes and axonal damage, which are key pathological features of MS. The study referenced (PLoS ONE 10 (4) (2015) e0119681) emphasizes the importance of CD163 in understanding the disease's progression and potentially guiding therapeutic strategies.\", text=Text(text='', name='Maroto2023 pages 12-12', doc=Doc(docname='Maroto2023', citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.', dockey='9dc57baad90f908d2c0e807487e6986a', overwrite_fields_from_metadata=True, formatted_citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.')), score=9), Context(context='The excerpt does not provide specific information about the involvement of CD163 in multiple sclerosis (MS). It primarily discusses the immunopathology of MS, focusing on the role of regulatory T cells (Treg cells) and their dysfunction in MS, as well as the environmental triggers that may contribute to the disease. The text mentions the migration of effector CD4+ T cells into the central nervous system (CNS) and their role in destroying myelin oligodendrocyte glycoprotein, but it does not mention CD163 or its role in MS. The focus is on the development of biomarkers for accurate diagnosis and the importance of early treatment to prevent disease progression.', text=Text(text='', name='Maroto2023 pages 1-2', doc=Doc(docname='Maroto2023', citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.', dockey='9dc57baad90f908d2c0e807487e6986a', overwrite_fields_from_metadata=True, formatted_citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.')), score=1), Context(context=\"The provided excerpt does not directly mention CD163 or its involvement in multiple sclerosis. Instead, it focuses on various aspects of multiple sclerosis research, including the role of regulatory T cells (Tregs), environmental risk factors, and the impact of Epstein-Barr virus on lymphoblastoid cell line survival. Specifically, it discusses the reduced suppressive effect of CD4+CD25high regulatory T cells on the T cell immune response against myelin oligodendrocyte glycoprotein in multiple sclerosis patients . Additionally, it mentions the role of IL-17-producing human peripheral regulatory T cells and their suppressive function . The excerpt also references studies on vitamin D's effect on multiple sclerosis relapses and the diagnosis and disease-modifying therapy for multiple sclerosis. However, none of these studies or references directly address CD163 or its role in multiple sclerosis.\", text=Text(text='', name='Maroto2023 pages 8-8', doc=Doc(docname='Maroto2023', citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.', dockey='9dc57baad90f908d2c0e807487e6986a', overwrite_fields_from_metadata=True, formatted_citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.')), score=1), Context(context=\"CD163, a macrophage scavenger receptor, is implicated in multiple sclerosis (MS) through its proteolytic shedding, which is observed in MS patients. This shedding process is linked to macrophage activity, as indicated by elevated levels of soluble CD163 in newly diagnosed MS patients. The presence of soluble CD163 serves as a marker of macrophage activity, suggesting its role in the inflammatory processes associated with MS. Additionally, CD163-positive B cell subsets have been identified in secondary progressive MS, further highlighting the involvement of CD163 in the disease's pathology. These findings underscore the potential of CD163 as a biomarker for monitoring disease activity and progression in MS.\", text=Text(text='', name='Maroto2023 pages 12-12', doc=Doc(docname='Maroto2023', citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.', dockey='9dc57baad90f908d2c0e807487e6986a', overwrite_fields_from_metadata=True, formatted_citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.')), score=9), Context(context='CD163, a hemoglobin-haptoglobin receptor, plays a significant role in iron uptake and storage in myeloid cells (MCs) within chronic active lesions in multiple sclerosis (MS). The study highlights that CD163 is involved in the active scavenging of iron, contributing to the formation of paramagnetic rim lesions (PRLs) observed in MS patients. Specifically, CD163+ cells were identified in lesion areas, and their expression correlated with iron accumulation at the rim of chronic active lesions. The study also found that MS patients with higher numbers of PRLs had elevated levels of soluble CD163 (sCD163) in cerebrospinal fluid (CSF), suggesting a link between CD163-mediated iron uptake and disease severity. Additionally, the research indicates that CD163, along with other iron import mechanisms, may be responsible for sustained iron storage in PRLs, emphasizing its role in the pathological processes underlying MS.', text=Text(text='', name='Hofmann2023 pages 9-10', doc=Doc(docname='Hofmann2023', citation='Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.', dockey='10b166c9c422b5fe2254ae8fb31f1adb', overwrite_fields_from_metadata=True, formatted_citation='Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.')), score=9), Context(context='CD163 plays a significant role in iron uptake and accumulation in myeloid cells (MCs) at the lesion rims of chronic active multiple sclerosis (MS) lesions. The study found that CD163, a scavenger receptor for hemoglobin-haptoglobin complexes, was highly expressed in MCs at these lesion rims, correlating strongly with iron accumulation . This suggests that CD163-mediated iron uptake is a key mechanism in MS pathology. Additionally, CD163 was part of the upregulated CD163-HMOX1-HAMP pathway, which is involved in iron metabolism and heme breakdown. CD163-expressing MCs also showed increased expression of IL-10 and C1QA, indicating a potential role in modulating inflammation and immune responses in MS lesions. These findings highlight CD163 as a critical player in iron-related pathology and immune regulation in MS.', text=Text(text='', name='Hofmann2023 pages 6-7', doc=Doc(docname='Hofmann2023', citation='Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.', dockey='10b166c9c422b5fe2254ae8fb31f1adb', overwrite_fields_from_metadata=True, formatted_citation='Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.')), score=10), Context(context=\"The provided excerpt does not directly mention CD163 or its involvement in multiple sclerosis. However, it discusses various biochemical and genetic factors related to multiple sclerosis, such as the role of HLA-DR2 in disease susceptibility, the impact of vitamin D on proinflammatory cytokine production, and the overactivity of MAPK/ERK in microglia. These factors are relevant to understanding the broader context of immune and inflammatory processes in multiple sclerosis, which could indirectly relate to CD163, a marker associated with anti-inflammatory macrophages. The excerpt also highlights environmental and lifestyle factors, genetic susceptibility, and immune system dysregulation in multiple sclerosis, which are critical areas of research in understanding the disease's pathogenesis.\", text=Text(text='', name='Maroto2023 pages 7-8', doc=Doc(docname='Maroto2023', citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.', dockey='9dc57baad90f908d2c0e807487e6986a', overwrite_fields_from_metadata=True, formatted_citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.')), score=3), Context(context='The excerpt does not provide any information about CD163 or its involvement in multiple sclerosis (MS). Instead, it discusses other biomarkers such as osteopontin (OPN), tubulin, heat shock protein 70 (HSP70), and chemokine ligand 13 (CXCL13) in the context of MS. These biomarkers are associated with inflammation, disease progression, and neuroinflammatory activity in MS. For example, OPN levels in CSF and blood are linked to MS progression, while CXCL13 is highlighted as a reliable biomarker for MS prognosis and neuroinflammatory activity. HSP70 is noted for its role in inflammation and its potential as a biomarker for monitoring MS. However, no mention of CD163 is made in the provided text.', text=Text(text='', name='Maroto2023 pages 6-6', doc=Doc(docname='Maroto2023', citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.', dockey='9dc57baad90f908d2c0e807487e6986a', overwrite_fields_from_metadata=True, formatted_citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.')), score=1), Context(context=\"The excerpt does not provide specific information about the role of CD163 in multiple sclerosis (MS). However, it discusses the broader context of MS as a chronic-inflammatory disease of the central nervous system, characterized by demyelination, axonal damage, and gliosis. The study focuses on creating a spatial map of gene expression across various inflammatory lesion stages in MS, using single-nucleus and spatial transcriptomics data. This approach aims to uncover patterns of glial, immune, and stromal cell subtype diversity, as well as cell-cell communication and signaling signatures across lesion and non-lesion tissue areas. The study highlights the importance of understanding the spatially restricted drivers of chronic tissue damage and lesion expansion in MS, which could lead to the identification of biomarkers and therapeutic targets. While CD163 is not mentioned, the study's focus on immune cell diversity and signaling in MS lesions may indirectly relate to the broader context of immune cell markers like CD163.\", text=Text(text='', name='Lerma2022 pages 1-3', doc=Doc(docname='Lerma2022', citation='Lerma-Martin, Celia, et al. \"Spatial Cell Type Mapping of Multiple Sclerosis Lesions.\" *bioRxiv*, 3 Nov. 2022, doi:10.1101/2022.11.03.514906. Accessed 2024.', dockey='31df84a215204ca9fb769ebc5ddcd0cb', overwrite_fields_from_metadata=True, formatted_citation='Lerma-Martin, Celia, et al. \"Spatial Cell Type Mapping of Multiple Sclerosis Lesions.\" *bioRxiv*, 3 Nov. 2022, doi:10.1101/2022.11.03.514906. Accessed 2024.')), score=3), Context(context=\"The provided excerpt does not contain specific information about the involvement of CD163 in multiple sclerosis. The text primarily discusses various aspects of multiple sclerosis, including diagnostic criteria, clinical course, treatment options, and biochemical pathways such as Wnt/β-catenin signaling and p38 mitogen-activated protein kinase. However, it does not mention CD163, a marker associated with anti-inflammatory macrophages, in the context of multiple sclerosis. Therefore, based on the given text, there is no direct information to answer the question about CD163's role in multiple sclerosis.\", text=Text(text='', name='Maroto2023 pages 7-7', doc=Doc(docname='Maroto2023', citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.', dockey='9dc57baad90f908d2c0e807487e6986a', overwrite_fields_from_metadata=True, formatted_citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.')), score=1), Context(context='In the context of multiple sclerosis (MS), CD163 plays a significant role in iron uptake and metabolism within myeloid cells (MCs) at the edges of chronic lesions, known as paramagnetic rim lesions (PRLs). The study by  highlights that iron uptake in these lesions is linked to rim-associated MC subtypes expressing C1QA, which are characterized by the upregulation of CD163, HMOX1, and HAMP, and downregulation of P2RY12. CD163 is a scavenger receptor for haptoglobin-bound hemoglobin, suggesting that haptoglobin-bound hemoglobin, rather than transferrin-bound iron, is a critical source for MC-associated iron uptake in MS. The study also found that people with MS (pwMS) who have higher numbers of PRLs exhibit elevated levels of soluble CD163 (sCD163) in their cerebrospinal fluid (CSF), correlating with the number of PRLs. This indicates that CD163-mediated iron uptake via haptoglobin-bound hemoglobin is relevant in MS pathology. Additionally, while haptoglobin (Hp) genotypes did not influence PRL counts, pwMS with Hp2-1/Hp2-2 haplotypes had higher clinical disability scores compared to those with Hp1-1, suggesting a potential role of CD163 in clinical worsening.', text=Text(text='', name='Hofmann2023 pages 1-2', doc=Doc(docname='Hofmann2023', citation='Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.', dockey='10b166c9c422b5fe2254ae8fb31f1adb', overwrite_fields_from_metadata=True, formatted_citation='Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.')), score=10), Context(context='The excerpt does not directly mention CD163 or its involvement in multiple sclerosis. However, it discusses various molecular and cellular mechanisms related to MS, such as oxidative stress, demyelination, and inflammation. For instance, it highlights the role of genes associated with oxidative stress in lesion progression and the adaptive response to demyelination. It also mentions the involvement of microglia, astrocytes, and macrophages in MS pathology, as well as the role of specific proteins like annexin A1 and raftlin in inflammation and immune responses. While CD163 is not explicitly discussed, its potential relevance could be inferred from its known role as a marker for anti-inflammatory macrophages, which may align with the broader context of immune regulation and inflammation in MS.', text=Text(text='', name='Alsema2024 pages 12-13', doc=Doc(docname='Alsema2024', citation='Alsema, Astrid M., et al. \"Spatially Resolved Gene Signatures of White Matter Lesion Progression in Multiple Sclerosis.\" *Nature Neuroscience*, vol. 27, no. 12, 2024, pp. 2341–2353. https://doi.org/10.1038/s41593-024-01765-6. Accessed 2024.', dockey='d20b9b4733a400aedae0a8d96c38f891', overwrite_fields_from_metadata=True, formatted_citation='Alsema, Astrid M., et al. \"Spatially Resolved Gene Signatures of White Matter Lesion Progression in Multiple Sclerosis.\" *Nature Neuroscience*, vol. 27, no. 12, 2024, pp. 2341–2353. https://doi.org/10.1038/s41593-024-01765-6. Accessed 2024.')), score=4), Context(context='The provided excerpt does not mention CD163 or its involvement in multiple sclerosis. The text primarily discusses biochemical biomarkers for multiple sclerosis, including vitamin D levels, oligoclonal IgM bands, and other biomarkers related to disease activity, progression, and cognitive dysfunction. It also references studies on cerebrospinal fluid (CSF) biomarkers, intrathecal IgG synthesis, and the role of the humoral immune system in multiple sclerosis. However, there is no specific information about CD163 or its role in the context of multiple sclerosis.', text=Text(text='', name='Maroto2023 pages 8-9', doc=Doc(docname='Maroto2023', citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.', dockey='9dc57baad90f908d2c0e807487e6986a', overwrite_fields_from_metadata=True, formatted_citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.')), score=1), Context(context='CD163 is a marker used in immunohistochemistry to study multiple sclerosis (MS) lesions. In the study, CD163 was stained using specific antibodies to identify its presence in postmortem human tissue samples from individuals diagnosed with MS. The tissue samples were obtained from the UK Multiple Sclerosis Tissue Bank and processed using immunohistochemistry techniques. CD163, along with other markers like CD68 and MOG, was used to assess histopathological changes in MS lesions. The staining process involved fixation, antibody application, and visualization using chromogens like DAB. CD163 is likely involved in the immune response within the MS lesion microenvironment, potentially playing a role in inflammation or macrophage activity, which are key aspects of MS pathology. However, the exact functional role of CD163 in MS is not explicitly detailed in the provided text, and further functional validation is suggested for future studies.', text=Text(text='', name='Lerma2022 pages 22-24', doc=Doc(docname='Lerma2022', citation='Lerma-Martin, Celia, et al. \"Spatial Cell Type Mapping of Multiple Sclerosis Lesions.\" *bioRxiv*, 3 Nov. 2022, doi:10.1101/2022.11.03.514906. Accessed 2024.', dockey='31df84a215204ca9fb769ebc5ddcd0cb', overwrite_fields_from_metadata=True, formatted_citation='Lerma-Martin, Celia, et al. \"Spatial Cell Type Mapping of Multiple Sclerosis Lesions.\" *bioRxiv*, 3 Nov. 2022, doi:10.1101/2022.11.03.514906. Accessed 2024.')), score=8), Context(context='The provided excerpt does not mention CD163 or its involvement in multiple sclerosis. The text primarily discusses the role of biomarkers such as IgG and IgM oligoclonal bands, the IgG index, and the use of MRI and McDonald criteria for diagnosing multiple sclerosis. It also mentions the importance of early therapeutic intervention and the regulation of MAPK pathways by DUSP. However, there is no information on CD163, a marker associated with macrophages and inflammation, in the context of multiple sclerosis.', text=Text(text='', name='Maroto2023 pages 2-3', doc=Doc(docname='Maroto2023', citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.', dockey='9dc57baad90f908d2c0e807487e6986a', overwrite_fields_from_metadata=True, formatted_citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.')), score=1), Context(context=\"The provided excerpt does not directly mention CD163 or its involvement in multiple sclerosis. It primarily discusses funding sources, acknowledgments, and references to other studies related to multiple sclerosis, including iron tracking, chronic active lesions, and inflammatory processes. The text highlights research grants and collaborations but does not provide specific information about CD163's role in the disease.\", text=Text(text='', name='Hofmann2023 pages 15-16', doc=Doc(docname='Hofmann2023', citation='Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.', dockey='10b166c9c422b5fe2254ae8fb31f1adb', overwrite_fields_from_metadata=True, formatted_citation='Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.')), score=1), Context(context='The provided excerpt does not mention CD163 or its involvement in multiple sclerosis (MS). The text focuses on the diagnosis and prognosis of MS using cerebrospinal fluid (CSF) and serum biomarkers, such as immunoglobulin G oligoclonal bands (OCB), kappa free light chains, and neurofilaments. It also discusses the importance of integrating clinical, imaging, and laboratory findings for MS diagnosis. However, there is no specific information about CD163, a marker associated with macrophages and inflammation, in the context of MS.', text=Text(text='', name='Maroto2023 pages 1-1', doc=Doc(docname='Maroto2023', citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.', dockey='9dc57baad90f908d2c0e807487e6986a', overwrite_fields_from_metadata=True, formatted_citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.')), score=1), Context(context=\"The excerpt does not directly mention CD163 or its involvement in multiple sclerosis (MS). However, it discusses the spatial mapping of cell types and signaling pathways in MS lesions, particularly focusing on the upregulation of TNF expression in MS tissues compared to controls. The study highlights the use of snRNA-seq and smFISH to analyze cell type-specific gene expression and dysregulation in MS, identifying patterns of cell-cell interaction and enrichment in specific tissue niches. It also mentions the mapping of MG/MΦ-MS subtypes to inflamed areas and the colocalization of certain cell types, such as SC, EC, and immune cell subtypes, with upregulated TGFβ signaling. While CD163 is not explicitly discussed, the findings on myeloid cell activation and perivascular immune cell cuffing may indirectly relate to CD163's role in MS pathology, as CD163 is a marker associated with macrophages and inflammation.\", text=Text(text='', name='Lerma2022 pages 19-20', doc=Doc(docname='Lerma2022', citation='Lerma-Martin, Celia, et al. \"Spatial Cell Type Mapping of Multiple Sclerosis Lesions.\" *bioRxiv*, 3 Nov. 2022, doi:10.1101/2022.11.03.514906. Accessed 2024.', dockey='31df84a215204ca9fb769ebc5ddcd0cb', overwrite_fields_from_metadata=True, formatted_citation='Lerma-Martin, Celia, et al. \"Spatial Cell Type Mapping of Multiple Sclerosis Lesions.\" *bioRxiv*, 3 Nov. 2022, doi:10.1101/2022.11.03.514906. Accessed 2024.')), score=4), Context(context='CD163 is a high-affinity scavenger receptor involved in the iron uptake and metabolism mechanisms in iron-laden myeloid cells (MCs) at chronic active multiple sclerosis (MS) lesion rims. It binds to haptoglobin-hemoglobin complexes at the surface of lesion rim-associated MCs, leading to the internalization of these complexes. This process is followed by the enzymatic breakdown of heme to form biliverdin through heme oxygenase (HMOX1) function, and the transport of ferrous iron (Fe2+) through DMT1 (encoded by SLC11A2). The iron can either be oxidized to Fe3+ and bound to ferritin or directly exported into extracellular spaces through ferroportin (FPN, encoded by SLC40A1), which is regulated by hepcidin (encoded by HAMP). HMOX1 is strongly upregulated in CD163 co-expressing MCs at chronic active lesion edges, indicating that CD163 is actively scavenging hemoglobin, a major source of iron in MCs at iron-accumulating MS lesion edges. Additionally, the upregulation of HAMP encoding hepcidin in MCs at MS lesion areas suggests that imported iron is actively retained in MCs, preventing its release into the surrounding tissue environment.', text=Text(text='', name='Hofmann2023 pages 13-15', doc=Doc(docname='Hofmann2023', citation='Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.', dockey='10b166c9c422b5fe2254ae8fb31f1adb', overwrite_fields_from_metadata=True, formatted_citation='Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.')), score=10), Context(context='The excerpt does not directly mention CD163 or its involvement in multiple sclerosis (MS). However, it discusses the roles of myeloid cell types, including microglia (MG) and macrophages (MΦ-MS), in MS lesions, particularly chronic active lesions. These cells exhibit transcriptional profiles associated with proinflammatory responses, such as upregulation of genes like CD84 and CD14, and downregulation of anti-inflammatory genes like INPP5D. Additionally, the study highlights the importance of transcription factors like SPI1 (PU.1) in MG cells, which are linked to MS. While CD163 is not explicitly mentioned, it is a marker often associated with anti-inflammatory macrophages, suggesting its potential relevance in the context of MS lesion dynamics and immune regulation.', text=Text(text='', name='Lerma2022 pages 12-15', doc=Doc(docname='Lerma2022', citation='Lerma-Martin, Celia, et al. \"Spatial Cell Type Mapping of Multiple Sclerosis Lesions.\" *bioRxiv*, 3 Nov. 2022, doi:10.1101/2022.11.03.514906. Accessed 2024.', dockey='31df84a215204ca9fb769ebc5ddcd0cb', overwrite_fields_from_metadata=True, formatted_citation='Lerma-Martin, Celia, et al. \"Spatial Cell Type Mapping of Multiple Sclerosis Lesions.\" *bioRxiv*, 3 Nov. 2022, doi:10.1101/2022.11.03.514906. Accessed 2024.')), score=4), Context(context='The excerpt does not explicitly mention CD163 or its involvement in multiple sclerosis (MS). However, it discusses the molecular and cellular processes in MS lesion rims, which are critical for understanding lesion expansion. The study uses spatial gene expression analysis to identify distinct gene signatures in white matter (WM) and grey matter (GM) MS lesions, including astrocyte-specific, oligodendrocyte-specific, and microglia-specific gene sets. These findings highlight the heterogeneity and dynamic progression of MS lesions, which may indirectly relate to the role of microglia and other immune cells, such as those expressing CD163, in MS pathology. CD163 is a marker for certain macrophage populations, which could be relevant to the inflammatory processes described in the study.', text=Text(text='', name='Alsema2024 pages 1-2', doc=Doc(docname='Alsema2024', citation='Alsema, Astrid M., et al. \"Spatially Resolved Gene Signatures of White Matter Lesion Progression in Multiple Sclerosis.\" *Nature Neuroscience*, vol. 27, no. 12, 2024, pp. 2341–2353. https://doi.org/10.1038/s41593-024-01765-6. Accessed 2024.', dockey='d20b9b4733a400aedae0a8d96c38f891', overwrite_fields_from_metadata=True, formatted_citation='Alsema, Astrid M., et al. \"Spatially Resolved Gene Signatures of White Matter Lesion Progression in Multiple Sclerosis.\" *Nature Neuroscience*, vol. 27, no. 12, 2024, pp. 2341–2353. https://doi.org/10.1038/s41593-024-01765-6. Accessed 2024.')), score=4), Context(context=\"The excerpt does not provide any information about CD163 or its involvement in multiple sclerosis. The text discusses various biomarkers such as S100β, Chitinase-3-like 1 (CHI3L1), CD5 + B cells, and osteopontin, but there is no mention of CD163. Therefore, based on the provided excerpt, there is no relevant information to answer the question about CD163's role in multiple sclerosis.\", text=Text(text='', name='Maroto2023 pages 5-6', doc=Doc(docname='Maroto2023', citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.', dockey='9dc57baad90f908d2c0e807487e6986a', overwrite_fields_from_metadata=True, formatted_citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.')), score=1), Context(context='CD163, a haptoglobin-hemoglobin receptor, is strongly enriched at the rims of chronic active multiple sclerosis (MS) lesions and is present on iron-laden myeloid cells (MCs). These MCs exhibit strong immunoreactivity for CD163, indicating its role in iron uptake and inflammation at lesion borders. CD163 is also associated with both pro-inflammatory (C1QA) and regulatory (IL10) functions, as its co-expression with C1QA is increased at lesion rims compared to lesion centers. This suggests that CD163 plays a dual role in MS pathology, contributing to both inflammatory and regulatory processes in spatially restricted lesion areas. Additionally, CD163 expression is linked to the accumulation of iron in MCs, which is a hallmark of chronic active MS lesions.', text=Text(text='', name='Hofmann2023 pages 7-9', doc=Doc(docname='Hofmann2023', citation='Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.', dockey='10b166c9c422b5fe2254ae8fb31f1adb', overwrite_fields_from_metadata=True, formatted_citation='Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.')), score=9), Context(context='The provided excerpt does not mention CD163 or its involvement in multiple sclerosis (MS). Instead, it focuses on the roles of microglia (MG) and macrophage (MΦ-MS) subtypes, their proinflammatory states, and their interactions with oligodendrocytes (OL) in MS lesions. Key findings include the upregulation of proinflammatory pathways (e.g., TNFα, JAK-STAT) and hypoxia-related pathways in MG states 2 and 3, as well as enriched ligand-receptor interactions such as APOE-TREM2 and APOE-LRP1, which are associated with myelin debris clearance and lipid metabolism in MS lesions. Additionally, the study highlights the transcriptomic heterogeneity of astrocyte (AS) subtypes and their roles in MS pathogenesis. However, no information is provided about CD163.', text=Text(text='', name='Lerma2022 pages 15-16', doc=Doc(docname='Lerma2022', citation='Lerma-Martin, Celia, et al. \"Spatial Cell Type Mapping of Multiple Sclerosis Lesions.\" *bioRxiv*, 3 Nov. 2022, doi:10.1101/2022.11.03.514906. Accessed 2024.', dockey='31df84a215204ca9fb769ebc5ddcd0cb', overwrite_fields_from_metadata=True, formatted_citation='Lerma-Martin, Celia, et al. \"Spatial Cell Type Mapping of Multiple Sclerosis Lesions.\" *bioRxiv*, 3 Nov. 2022, doi:10.1101/2022.11.03.514906. Accessed 2024.')), score=1), Context(context='The provided excerpt does not mention CD163 or its involvement in multiple sclerosis. Instead, it focuses on biochemical biomarkers such as IgG intrathecal synthesis, kappa and lambda free light chains, and neurofilaments in the context of MS diagnosis and prognosis. IgG oligoclonal bands (OCB) in cerebrospinal fluid (CSF) are highlighted as a gold standard for detecting intrathecal synthesis, while kappa and lambda free light chains are discussed as markers of B-lymphocyte activity in inflammatory CNS disorders. Neurofilaments, particularly NfL, are described as biomarkers for axonal damage in MS and other neurological conditions. The excerpt also references the McDonald Criteria 2017 for MS diagnosis. Since CD163 is not mentioned, there is no relevant information about its role in MS in this text.', text=Text(text='', name='Maroto2023 pages 3-4', doc=Doc(docname='Maroto2023', citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.', dockey='9dc57baad90f908d2c0e807487e6986a', overwrite_fields_from_metadata=True, formatted_citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.')), score=1), Context(context=\"CD163, specifically soluble CD163 (sCD163), is analyzed in cerebrospinal fluid (CSF) samples obtained at the time of multiple sclerosis (MS) diagnosis. The concentration of sCD163 is measured using the ultrasensitive Ella-Simple Plex technology, an automated ELISA method, to assess its levels in relation to MS. This analysis is performed by independent investigators who are blinded to the patients' clinical and MRI features, ensuring unbiased results. The reproducibility and consistency of the sCD163 measurements are verified by running all samples in triplicates. The involvement of CD163 in MS is likely related to its role in the immune response and inflammation, as it is a marker associated with macrophages and monocytes, which are implicated in the pathogenesis of MS.\", text=Text(text='', name='Hofmann2023 pages 4-5', doc=Doc(docname='Hofmann2023', citation='Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.', dockey='10b166c9c422b5fe2254ae8fb31f1adb', overwrite_fields_from_metadata=True, formatted_citation='Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.')), score=8), Context(context='The provided excerpt does not mention CD163 or its involvement in multiple sclerosis. Instead, it focuses on the role of vitamin D in multiple sclerosis, discussing its association with relapse rates, disability, and clinical outcomes. It also mentions the role of Tregs (regulatory T cells) and their secretion of IL-17, as well as the expression of the TH17 lineage-specific transcription factor RORγt. Additionally, the excerpt references studies on oligoclonal bands in cerebrospinal fluid and IgG synthesis within the central nervous system, which are biomarkers for multiple sclerosis. However, there is no direct information about CD163 in this context.', text=Text(text='', name='Maroto2023 pages 8-8', doc=Doc(docname='Maroto2023', citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.', dockey='9dc57baad90f908d2c0e807487e6986a', overwrite_fields_from_metadata=True, formatted_citation='Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.')), score=1)], references='1. (Hofmann2023 pages 1-2): Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.\\n\\n2. (Hofmann2023 pages 10-12): Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.\\n\\n3. (Hofmann2023 pages 13-15): Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.\\n\\n4. (Hofmann2023 pages 6-7): Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.\\n\\n5. (Hofmann2023 pages 7-9): Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.\\n\\n6. (Maroto2023 pages 6-7): Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.\\n\\n7. (Hofmann2023 pages 9-10): Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.', formatted_answer='Question: What is the gene CD163? And how is CD163 involved in multiple sclerosis?\\n\\n**CD163** is a gene encoding a scavenger receptor that binds to the hemoglobin-haptoglobin complex, playing a critical role in iron uptake and metabolism within myeloid cells (MCs), particularly in the context of inflammatory and neurodegenerative diseases such as multiple sclerosis (MS) (Hofmann2023 pages 1-2, Hofmann2023 pages 10-12). CD163 is primarily expressed on perivascular macrophages in the healthy brain but is upregulated on resident microglia and infiltrating macrophages under pathological conditions, including chronic active MS lesions, HIV encephalitis, and subarachnoid hemorrhage (Hofmann2023 pages 10-12). This receptor facilitates the internalization of haptoglobin-bound hemoglobin, which is enzymatically broken down by heme oxygenase (HMOX1) to release ferrous iron (Fe2+). The iron is then either stored in ferritin or exported via ferroportin (FPN), regulated by hepcidin (HAMP) (Hofmann2023 pages 13-15).\\n\\nIn MS, CD163 is significantly upregulated on a subtype of MCs at the rims of chronic active lesions, known as paramagnetic rim lesions (PRLs), where it correlates strongly with iron-positive MCs (Hofmann2023 pages 6-7, Hofmann2023 pages 10-12). This upregulation is associated with pro-inflammatory activation, as indicated by the co-expression of markers like C1QA and HLA-DR, and is linked to compromised blood-brain barrier integrity, allowing the influx of haptoglobin-hemoglobin complexes into the central nervous system (CNS) (Hofmann2023 pages 10-12, Hofmann2023 pages 7-9). The CD163-HMOX1-HAMP pathway, which is upregulated in MCs at chronic active lesion rims, suggests a mechanism for iron uptake and regulation independent of transferrin-mediated pathways (Hofmann2023 pages 6-7).\\n\\nCD163-mediated iron uptake contributes to iron accumulation at lesion rims, a hallmark of chronic active MS lesions, and is implicated in the formation of PRLs (Hofmann2023 pages 9-10). Elevated levels of soluble CD163 (sCD163), shed by ADAM17, are observed in the cerebrospinal fluid (CSF) of MS patients with higher PRL counts, correlating with disease severity and clinical disability (Hofmann2023 pages 1-2, Hofmann2023 pages 9-10). Additionally, CD163-expressing MCs exhibit both pro-inflammatory and regulatory functions, with increased co-expression of C1QA at lesion rims and IL-10 in peri-plaque white matter, highlighting its dual role in MS pathology (Hofmann2023 pages 7-9).\\n\\nIn summary, CD163 is a key player in iron metabolism and immune regulation in MS, contributing to iron accumulation, inflammation, and lesion progression. Its soluble form, sCD163, has been proposed as a potential biomarker for MS lesion activity and clinical disability (Hofmann2023 pages 10-12, Maroto2023 pages 6-7).\\n\\nReferences\\n\\n1. (Hofmann2023 pages 1-2): Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.\\n\\n2. (Hofmann2023 pages 10-12): Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.\\n\\n3. (Hofmann2023 pages 13-15): Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.\\n\\n4. (Hofmann2023 pages 6-7): Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.\\n\\n5. (Hofmann2023 pages 7-9): Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.\\n\\n6. (Maroto2023 pages 6-7): Maroto-García, Julia, et al. \"Biochemical Biomarkers for Multiple Sclerosis.\" *Clinica Chimica Acta*, vol. 548, 2023, p. 117471. ScienceDirect, www.elsevier.com/locate/cca. Accessed 2024.\\n\\n7. (Hofmann2023 pages 9-10): Hofmann, Annika, et al. \"Myeloid Cell Iron Uptake Pathways and Paramagnetic Rim Formation in Multiple Sclerosis.\" *Acta Neuropathologica*, vol. 146, 2023, pp. 707–724. Springer, https://doi.org/10.1007/s00401-023-02627-4. Accessed 2024.\\n', graded_answer=None, cost=0.015463980000000004, token_counts={'deepseek/deepseek-chat': [88699, 10879]}, config_md5='358b29b73eb9598e5ad6bb0479937e42', tool_history=[['paper_search', 'paper_search'], ['gather_evidence', 'gather_evidence'], ['gen_answer'], ['complete']], used_contexts={'Hofmann2023 pages 7-9', 'Maroto2023 pages 6-7', 'Hofmann2023 pages 6-7', 'Hofmann2023 pages 1-2', 'Hofmann2023 pages 13-15', 'Hofmann2023 pages 9-10', 'Hofmann2023 pages 10-12'}), bibtex=None, status=<AgentStatus.SUCCESS: 'success'>, timing_info=None, duration=0.0, stats=None)"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query_answer_index_results = await query_answer_index(settings=settings, query=\"CD163\")\n",
    "query_answer_index_results[0]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Appendix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from paperqa.utils import parse_string, clean_upbibtex, unsrtalpha, CitationConversionError, Person, Parser, FieldIsMissing\n",
    "\n",
    "\n",
    "def format_bibtex(\n",
    "    bibtex: str,\n",
    "    key: str | None = None,\n",
    "    clean: bool = True,\n",
    "    missing_replacements: dict[str, str] | None = None,\n",
    ") -> str:\n",
    "    \"\"\"Transform bibtex entry into a citation, potentially adding missing fields.\"\"\"\n",
    "    style = unsrtalpha.Style()\n",
    "    if missing_replacements is None:\n",
    "        missing_replacements = {}\n",
    "    try:\n",
    "        bd = parse_string(clean_upbibtex(bibtex) if clean else bibtex, \"bibtex\")\n",
    "        key = list(bd.entries.keys())[0]\n",
    "    except Exception:\n",
    "        key = bibtex.split(\",\")[0]\n",
    "        key = key.replace(\"{{\", \"{\")\n",
    "        key = key.split(\"{\")[1]\n",
    "        return \"Ref \" + key\n",
    "    try:\n",
    "        entry = bd.entries[key]\n",
    "    except KeyError as exc:  # Let's check if key is a non-empty prefix\n",
    "        try:\n",
    "            entry = next(\n",
    "                iter(v for k, v in bd.entries.items() if k.startswith(key) and key)\n",
    "            )\n",
    "        except StopIteration:\n",
    "            raise CitationConversionError(\n",
    "                f\"Failed to process{' and clean up' if clean else ''} bibtex {bibtex}\"\n",
    "                f\" due to failed lookup of key {key}.\"\n",
    "            ) from exc\n",
    "    try:\n",
    "        # see if we can insert missing fields\n",
    "        for field, replacement_value in missing_replacements.items():\n",
    "            # Deal with special case for author, since it needs to be parsed\n",
    "            # into Person objects. This reorganizes the names automatically.\n",
    "            if field == \"author\" and \"author\" not in entry.persons:\n",
    "                tmp_author_bibtex = f\"@misc{{tmpkey, author={{{replacement_value}}}}}\"\n",
    "                authors: list[Person] = (\n",
    "                    Parser()\n",
    "                    .parse_string(tmp_author_bibtex)\n",
    "                    .entries[\"tmpkey\"]\n",
    "                    .persons[\"author\"]\n",
    "                )\n",
    "                for a in authors:\n",
    "                    entry.add_person(a, \"author\")\n",
    "            elif field not in entry.fields:\n",
    "                entry.fields.update({field: replacement_value})\n",
    "        entry = style.format_entry(label=\"1\", entry=entry)\n",
    "        return entry.text.render_as(\"text\")\n",
    "    except (FieldIsMissing, UnicodeDecodeError):\n",
    "        try:\n",
    "            return entry.fields[\"title\"]\n",
    "        except KeyError as exc:\n",
    "            raise CitationConversionError(\n",
    "                f\"Failed to process{' and clean up' if clean else ''} bibtex {bibtex}\"\n",
    "                \" due to missing a 'title' field.\"\n",
    "            ) from exc\n",
    "    \n",
    "missing_replacements = None\n",
    "clean = True\n",
    "bibtex = \"\"\"@article{{tortosacarreres}2024predictivepotentialof,\n",
    "    author = \"{Tortosa-Carreres}, Jordi and {Cubas-N{\\'u}{\\\\textasciitilde n}ez}, Laura and {Quiroga-Varela}, Ana and {Castillo-Villalba}, Jessica and {Rami{\\'o}-Torrenta}, Llu{\\'i}s and Piqueras, M{\\'o}nica and {Gasqu{\\'e}-Rubio}, Raquel and {Quintanilla-Bordas}, Carlos and Sanz, Maria Teresa and Lucas, Celia and {Huertas-Pons}, Joana Mar{\\'i}a and Miguela, Albert and Casanova, Bonaventura and {Laiz-Marro}, Bego{\\\\textasciitilde n}a and {P{\\'e}rez-Miralles}, Francisco Carlos\",\n",
    "    title = \"Predictive Potential of Serum and Cerebrospinal Fluid Biomarkers for Disease Activity in Treated Multiple Sclerosis Patients\",\n",
    "    year = \"2024\",\n",
    "    journal = \"Multiple Sclerosis and Related Disorders\",\n",
    "    doi = \"10.1016/j.msard.2024.105734\",\n",
    "    url = \"https://doi.org/10.1016/j.msard.2024.105734\",\n",
    "    publisher = \"Elsevier\",\n",
    "    issn = \"2211-0348, 2211-0356\"\n",
    "}\n",
    "\"\"\"\n",
    "\n",
    "\n",
    "\"\"\"Transform bibtex entry into a citation, potentially adding missing fields.\"\"\"\n",
    "style = unsrtalpha.Style()\n",
    "if missing_replacements is None:\n",
    "    missing_replacements = {}\n",
    "try:\n",
    "    print(bibtex)\n",
    "    bd = parse_string(clean_upbibtex(bibtex) if clean else bibtex, \"bibtex\")\n",
    "    key = list(bd.entries.keys())[0]\n",
    "except Exception:\n",
    "    key = bibtex.split(\",\")[0]\n",
    "    key = key.replace(\"{{\", \"{\")\n",
    "    key = key.split(\"{\")[1]\n",
    "    print(\"Ref \" + key)\n",
    "try:\n",
    "    entry = bd.entries[key]\n",
    "    print(entry)\n",
    "except KeyError as exc:  # Let's check if key is a non-empty prefix\n",
    "    try:\n",
    "        entry = next(\n",
    "            iter(v for k, v in bd.entries.items() if k.startswith(key) and key)\n",
    "        )\n",
    "    except StopIteration:\n",
    "        raise CitationConversionError(\n",
    "            f\"Failed to process{' and clean up' if clean else ''} bibtex {bibtex}\"\n",
    "            f\" due to failed lookup of key {key}.\"\n",
    "        ) from exc\n",
    "try:\n",
    "    # see if we can insert missing fields\n",
    "    for field, replacement_value in missing_replacements.items():\n",
    "        # Deal with special case for author, since it needs to be parsed\n",
    "        # into Person objects. This reorganizes the names automatically.\n",
    "        if field == \"author\" and \"author\" not in entry.persons:\n",
    "            tmp_author_bibtex = f\"@misc{{tmpkey, author={{{replacement_value}}}}}\"\n",
    "            authors: list[Person] = (\n",
    "                Parser()\n",
    "                .parse_string(tmp_author_bibtex)\n",
    "                .entries[\"tmpkey\"]\n",
    "                .persons[\"author\"]\n",
    "            )\n",
    "            for a in authors:\n",
    "                entry.add_person(a, \"author\")\n",
    "        elif field not in entry.fields:\n",
    "            entry.fields.update({field: replacement_value})\n",
    "    entry = style.format_entry(label=\"1\", entry=entry)\n",
    "    print(entry.text.render_as(\"text\"))\n",
    "except (FieldIsMissing, UnicodeDecodeError):\n",
    "    try:\n",
    "        print(entry.fields[\"title\"])\n",
    "    except KeyError as exc:\n",
    "        raise CitationConversionError(\n",
    "            f\"Failed to process{' and clean up' if clean else ''} bibtex {bibtex}\"\n",
    "            \" due to missing a 'title' field.\"\n",
    "        ) from exc\n",
    "\n",
    "\n",
    "#test = \"\"\"@article{{tortosacarreres}2024predictivepotentialof,\n",
    "#    author = \"{Tortosa-Carreres}, Jordi and {Cubas-N{\\'u}{\\\\textasciitilde n}ez}, Laura and {Quiroga-Varela}, Ana and {Castillo-Villalba}, Jessica and {Rami{\\'o}-Torrenta}, Llu{\\'i}s and Piqueras, M{\\'o}nica and {Gasqu{\\'e}-Rubio}, Raquel and {Quintanilla-Bordas}, Carlos and Sanz, Maria Teresa and Lucas, Celia and {Huertas-Pons}, Joana Mar{\\'i}a and Miguela, Albert and Casanova, Bonaventura and {Laiz-Marro}, Bego{\\\\textasciitilde n}a and {P{\\'e}rez-Miralles}, Francisco Carlos\",\n",
    "#    title = \"Predictive Potential of Serum and Cerebrospinal Fluid Biomarkers for Disease Activity in Treated Multiple Sclerosis Patients\",\n",
    "#    year = \"2024\",\n",
    "#    journal = \"Multiple Sclerosis and Related Disorders\",\n",
    "#    doi = \"10.1016/j.msard.2024.105734\",\n",
    "#    url = \"https://doi.org/10.1016/j.msard.2024.105734\",\n",
    "#    publisher = \"Elsevier\",\n",
    "#    issn = \"2211-0348, 2211-0356\"\n",
    "#\"\"\"\n",
    "#format_bibtex(test)\n",
    "#parse_string(clean_upbibtex(test), \"bibtex\")\n",
    "\n",
    "bd = parse_string(clean_upbibtex(bibtex) if clean else bibtex, \"bibtex\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(bibtex)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "test = \"\"\"@article{tortosacarreres2024predictivepotentialof,\n",
    "  author = \"Tortosa-Carreres, Jordi and Cubas-Núñez, Laura and Quiroga-Varela, Ana and Castillo-Villalba, Jessica and Ramió-Torrenta, Lluís and Piqueras, Mónica and Gasqué-Rubio, Raquel and Quintanilla-Bordas, Carlos and Sanz, Maria Teresa and Lucas, Celia and Huertas-Pons, Joana María and Miguela, Albert and Casanova, Bonaventura and Laiz-Marro, Begoña and Pérez-Miralles, Francisco Carlos\",\n",
    "  title = \"Predictive Potential of Serum and Cerebrospinal Fluid Biomarkers for Disease Activity in Treated Multiple Sclerosis Patients\",\n",
    "  year = \"2024\",\n",
    "  journal = \"Multiple Sclerosis and Related Disorders\",\n",
    "  doi = \"10.1016/j.msard.2024.105734\",\n",
    "  url = \"https://doi.org/10.1016/j.msard.2024.105734\",\n",
    "  publisher = \"Elsevier\",\n",
    "  issn = \"2211-0348, 2211-0356\"\n",
    "}\"\"\"\n",
    "parse_string(test, bib_format=\"bibtex\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "test = \"\"\"@article{tortosacarreres2024predictivepotentialof,\n",
    "    author = \"foo, bar\",\n",
    "    title = \"Predictive Potential of Serum and Cerebrospinal Fluid Biomarkers for Disease Activity in Treated Multiple Sclerosis Patients\",\n",
    "    year = \"2024\",\n",
    "    journal = \"Multiple Sclerosis and Related Disorders\",\n",
    "    doi = \"10.1016/j.msard.2024.105734\",\n",
    "    url = \"https://doi.org/10.1016/j.msard.2024.105734\",\n",
    "    publisher = \"Elsevier\",\n",
    "    issn = \"2211-0348\"\n",
    "}\"\"\"\n",
    "parse_string(test, bib_format=\"bibtex\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "DocDetails(**doc_details_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pybtex\n",
    "\n",
    "bib_file = "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "entry"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "clean = False\n",
    "bd = parse_string(clean_upbibtex(test) if clean else test, \"bibtex\")\n",
    "list(bd.entries.keys())[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "bd = parse_string(clean_upbibtex(bibtex) if clean else bibtex, \"bibtex\")\n",
    "bd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "paperqa-env",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
